---
output: github_document
---

<!-- README.md is generated from README.Rmd. Please edit that file -->

```{r, include = FALSE}
knitr::opts_chunk$set(
    collapse = TRUE,
    comment = "#>",
    fig.path = "man/figures/README-",
    out.width = "100%"
)
```

# qsvaRPkgDemo

<!-- badges: start -->
[![GitHub issues](https://img.shields.io/github/issues/HediaTnani/qsvaRPkgDemo)](https://github.com/HediaTnani/qsvaRPkgDemo/issues)
[![GitHub pulls](https://img.shields.io/github/issues-pr/HediaTnani/qsvaRPkgDemo)](https://github.com/HediaTnani/qsvaRPkgDemo/pulls)
<!-- badges: end -->

The goal of `qsvaRPkgDemo` is to provide a step-by-step explanation of the qsvaR package for entry-level users and apply it to BrainSeq Phase 2 dataset. We'll also show how qsvaR adjust for the effect of degradation in RNAseq postmortem human brain  datasets. Using qsvaR can improve significantly the reproducibility of differential expression analyses across datasets.

## Installation instructions

Get the latest stable `R` release from [CRAN](http://cran.r-project.org/). Then install `qsvaRPkgDemo` from [Bioconductor](http://bioconductor.org/) using the following code:

```{r 'install', eval = FALSE}
if (!requireNamespace("BiocManager", quietly = TRUE)) {
    install.packages("BiocManager")
}

BiocManager::install("qsvaRPkgDemo")
```

And the development version from [GitHub](https://github.com/HediaTnani/qsvaRPkgDemo) with:

```{r 'install_dev', eval = FALSE}
BiocManager::install("HediaTnani/qsvaRPkgDemo")
```


## Citation

Below is the citation output from using `citation('qsvaRPkgDemo')` in R. Please
run this yourself to check for any updates on how to cite __qsvaRPkgDemo__.

```{r 'citation', eval = requireNamespace('qsvaRPkgDemo')}
print(citation('qsvaRPkgDemo'), bibtex = TRUE)
```

Please note that the `qsvaRPkgDemo` was only made possible thanks to many other R and bioinformatics software authors, which are cited either in the vignettes and/or the paper(s) describing this package.

## Code of Conduct

Please note that the `qsvaRPkgDemo` project is released with a [Contributor Code of Conduct](http://bioconductor.org/about/code-of-conduct/). By contributing to this project, you agree to abide by its terms.

## Development tools

* Continuous code testing is possible thanks to [GitHub actions](https://www.tidyverse.org/blog/2020/04/usethis-1-6-0/)  through `r BiocStyle::CRANpkg('usethis')`, `r BiocStyle::CRANpkg('remotes')`, and `r BiocStyle::CRANpkg('rcmdcheck')` customized to use [Bioconductor's docker containers](https://www.bioconductor.org/help/docker/) and `r BiocStyle::Biocpkg('BiocCheck')`.
* Code coverage assessment is possible thanks to [codecov](https://codecov.io/gh) and `r BiocStyle::CRANpkg('covr')`.
* The [documentation website](http://HediaTnani.github.io/qsvaRPkgDemo) is automatically updated thanks to `r BiocStyle::CRANpkg('pkgdown')`.
* The code is styled automatically thanks to `r BiocStyle::CRANpkg('styler')`.
* The documentation is formatted thanks to `r BiocStyle::CRANpkg('devtools')` and `r BiocStyle::CRANpkg('roxygen2')`.

For more details, check the `dev` directory.

This package was developed using `r BiocStyle::Biocpkg('biocthis')`.


