<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Differential Expression Analysis using Limma and qsvaR • qsvaRPkgDemo</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet">
<meta property="og:title" content="Differential Expression Analysis using Limma and qsvaR">
<meta property="og:description" content="qsvaRPkgDemo">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">qsvaRPkgDemo</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.99.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/qsvaRPkgDemo.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/HediaTnani/qsvaRPkgDemo/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Differential Expression Analysis using Limma and
qsvaR</h1>
                        <h4 data-toc-skip class="author">Hedia
Tnani</h4>
            <address class="author_afil">
      Lieber Institute for Brain Development, Johns Hopkins Medical
Campus<br><a class="author_email" href="mailto:#"></a><a href="mailto:hediatnani0@gmail.com" class="email">hediatnani0@gmail.com</a>
      </address>
                              <h4 data-toc-skip class="author">Joshua M.
Stolz</h4>
            <address class="author_afil">
      Lieber Institute for Brain Development, Johns Hopkins Medical
Campus<br><a class="author_email" href="mailto:#"></a><a href="mailto:jstolz80@gmail.com" class="email">jstolz80@gmail.com</a>
      </address>
                              <h4 data-toc-skip class="author">Louise A.
Huuki-Myers</h4>
            <address class="author_afil">
      Lieber Institute for Brain Development, Johns Hopkins Medical
Campus<br><a class="author_email" href="mailto:#"></a><a href="mailto:lahuuki@gmail.com" class="email">lahuuki@gmail.com</a>
      </address>
                              <h4 data-toc-skip class="author">Leonardo
Collado-Torres</h4>
            <address class="author_afil">
      Lieber Institute for Brain Development, Johns Hopkins Medical
Campus<br><a class="author_email" href="mailto:#"></a><a href="mailto:lcolladotor@gmail.com" class="email">lcolladotor@gmail.com</a>
      </address>
                  
            <h4 data-toc-skip class="date">25 July 2023</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/HediaTnani/qsvaRPkgDemo/blob/HEAD/vignettes/qsvaRPkgDemo.Rmd" class="external-link"><code>vignettes/qsvaRPkgDemo.Rmd</code></a></small>
      <div class="hidden name"><code>qsvaRPkgDemo.Rmd</code></div>

    </div>

    
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>Postmortem brain tissue is a valuable resource for studying the human
brain and understanding various neurological and psychiatric disorders.
However, the quality of postmortem brain tissue can be significantly
affected by degradation, which can occur due to various factors such as
the time elapsed since death (postmortem interval), the conditions in
which the body was stored, and the method of tissue preservation.</p>
<p>Therefore, it’s crucial to remove or at least minimize the effect of
degradation in postmortem brain tissue to ensure the accuracy and
reliability of research findings. This can be achieved through various
methods, such as optimizing the conditions for tissue storage and
preservation, using appropriate controls, and using statistical methods
to correct for the effects of degradation.</p>
<p>This is a basic example which shows how to obtain the quality
surrogate variables (qSVs) for BrainSeq Phase 2.</p>
</div>
<div class="section level2">
<h2 id="rbioconductor-packages-used">R/Bioconductor packages used<a class="anchor" aria-label="anchor" href="#rbioconductor-packages-used"></a>
</h2>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://bioconductor.org/packages/SummarizedExperiment" class="external-link">SummarizedExperiment</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/stemangiola/tidySummarizedExperiment" class="external-link">tidySummarizedExperiment</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://tidyverse.tidyverse.org" class="external-link">tidyverse</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="http://bioconductor.org/packages/variancePartition" class="external-link">variancePartition</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/jokergoo/ComplexHeatmap" class="external-link">ComplexHeatmap</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/jokergoo/colorRamp2" class="external-link">colorRamp2</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://magrittr.tidyverse.org" class="external-link">magrittr</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://purrr.tidyverse.org/" class="external-link">purrr</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="http://bioinf.wehi.edu.au/limma" class="external-link">limma</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/LieberInstitute/qsvaR" class="external-link">qsvaR</a></span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="time-outline">Time Outline<a class="anchor" aria-label="anchor" href="#time-outline"></a>
</h2>
<table class="table">
<thead><tr class="header">
<th align="center">Activity</th>
<th align="center">Time</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="center">Introduction to qsvaR (slides)</td>
<td align="center">10m</td>
</tr>
<tr class="even">
<td align="center">Data overview</td>
<td align="center">5m</td>
</tr>
<tr class="odd">
<td align="center">Limma and qsvaR</td>
<td align="center">20m</td>
</tr>
</tbody>
</table>
<p>Total: a 35 minute session.</p>
</div>
<div class="section level2">
<h2 id="workshop-goals">Workshop goals<a class="anchor" aria-label="anchor" href="#workshop-goals"></a>
</h2>
<p>During the workshop, we will provide a step-by-step explanation of
the qsvaR for entry-level users and apply it to BrainSeq Phase 2
dataset. We will describe how using qsvaR can improve the
reproducibility of DE analyses across datasets.</p>
</div>
<div class="section level2">
<h2 id="brainseq-phase-2-dataset">BrainSeq Phase 2 Dataset<a class="anchor" aria-label="anchor" href="#brainseq-phase-2-dataset"></a>
</h2>
<p>For illustrative purposes, we’ll use the BrainSeq Phase 2 data at the
transcript level <span class="citation">[@Collado-Torres2019]</span>.
This dataset is RiboZero RNA-seq data for 900 samples across both the
dorsolateral prefrontal cortex (DLPFC) and the hippocampus (HIPPO) for
551 individuals (286 affected by schizophrenia disorder: SCZD) . Visit
<a href="https://eqtl.brainseq.org/phase2/" class="external-link">here</a> for more
details.</p>
<div class="section level3">
<h3 id="load-the-data">Load the data<a class="anchor" aria-label="anchor" href="#load-the-data"></a>
</h3>
<p>The BrainSeq Phase 2 data at the transcript level resides in a
<code>RangedSummarizedExperiment</code> (RSE) object called
<code>rse_tx</code>.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Load the rse_tx object</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/load.html" class="external-link">load</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/connections.html" class="external-link">url</a></span><span class="op">(</span><span class="st">"https://s3.us-east-2.amazonaws.com/libd-brainseq2/rse_tx_unfiltered.Rdata"</span><span class="op">)</span>, verbose <span class="op">=</span> <span class="cn">T</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## General overview of the object</span></span>
<span><span class="va">rse_tx</span></span></code></pre></div>
<p>This is a <code>SummarizedExperiment</code> object but it is
evaluated as a tibble. So it is fully compatible both with
<code>SummarizedExperiment</code> and tidyverse APIs.</p>
</div>
<div class="section level3">
<h3 id="summarizedexperiment">SummarizedExperiment<a class="anchor" aria-label="anchor" href="#summarizedexperiment"></a>
</h3>
<p>The <code>SummarizedExperiment</code> package contains two classes:
<code>SummarizedExperiment</code> and
<code>RangedSummarizedExperiment</code>.</p>
<p><code>RangedSummarizedExperiment</code> is a subclass of the
SummarizedExperiment class, meaning it inherits all the features and
functionalities of its parent class. The primary difference is that the
rows of a <code>RangedSummarizedExperiment</code> object represent
genomic ranges of interest, rather than a <code>DataFrame</code> of
features. Here’s what you need to know about
RangedSummarizedExperiment:</p>
<p><code>GenomicRanges</code> Representation:
<code>RangedSummarizedExperiment</code> objects use a
<code>GRanges</code> or <code>GRangesList</code> object to represent
genomic ranges. These objects store information about genomic
coordinates, genomic features, and annotations for the ranges of
interest. The <code><a href="https://rdrr.io/pkg/MatrixGenerics/man/rowRanges.html" class="external-link">rowRanges()</a></code> function allows you to access and
manipulate the genomic range information within a
<code>RangedSummarizedExperiment</code> object.</p>
<p>The following graphic displays the class geometry and highlights the
vertical (column) and horizontal (row) relationships.</p>
<div class="float">
<img src="figs/rse_object.png" style="width:400.0%" alt="RangedSummarizedExperiment"><div class="figcaption">RangedSummarizedExperiment</div>
</div>
</div>
<div class="section level3">
<h3 id="sample-data">Sample data<a class="anchor" aria-label="anchor" href="#sample-data"></a>
</h3>
<p><code>SAMPLE_ID</code> : is the name of the sample.</p>
<p><code>ERCCsumLogErr</code> : a summary statistic quantifying overall
difference of expected and actual ERCC concentrations for one
sample.</p>
<p><code>numReads</code> : Total number of reads in fastq file.</p>
<p><code>numMapped</code> : Number of reads that were aligned by HISAT2
to the reference genome.</p>
<p><code>numUnmapped</code> : Number of reads that did not align to the
reference genome.</p>
<p><code>overallMapRate</code> : Proportion of total reads mapped
(numMapped / numReads).</p>
<p><code>concordMapRate</code> : If library is paired-end, percentage of
read pairs that mapped concordantly.</p>
<p><code>overallMapRate</code>: the decimal fraction of reads which
successfully mapped to the reference genome
(i.e.numMapped/numReads).</p>
<p><code>totalMapped</code>: the number of reads which successfully
mapped to the canonical sequences in the reference genome (excluding
mitochondrial chromosomes).</p>
<p><code>mitoMapped</code>: the number of reads which successfully
mapped to the mitochondrial chromosome.</p>
<p><code>mitoRate</code> : the decimal fraction of reads which mapped to
the mitochondrial chromosome, of those which map at all (i.e.mitoMapped/
(totalMapped+mitoMapped)).</p>
<p><code>totalAssignedGene</code> : the decimal fraction of reads
assigned unambiguously to a gene (including mitochondrial genes), with
featureCounts(Liao et al.2014), of those in total.</p>
<p><code>rRNA_rate</code> : the decimal fraction of reads assigned to a
gene whose type is ‘rRNA’, of those assigned to any gene.</p>
<p><code>RNum</code> : stands for Sample ID.</p>
<p><code>BrNum</code> : stands for Brain Number. It likely refers to a
unique identifier assigned to each brain (or brain sample) examined in
the study.</p>
<p><code>Region</code> : refers to a specific region of the brain from
which the sample was taken.</p>
<p><code>RIN</code> : refers to RNA Integrity Number. It’s a measure of
the quality of the RNA samples used in the study. A higher RIN indicates
higher quality, and thus more reliable, RNA.</p>
<p><code>Age</code> : refers to the age of the donors from which the
samples were taken.</p>
<p><code>Sex</code> : refers to the sex of the donors from which the
samples were taken. if the sample comes from a female (F) or a male (M)
donor.</p>
<p><code>Race</code> : refers to the race of the donors from which the
samples were taken.</p>
<p><code>Dx</code> : refers to the diagnosis of the donors from which
the samples were taken.</p>
</div>
<div class="section level3">
<h3 id="data-normalization">Data normalization<a class="anchor" aria-label="anchor" href="#data-normalization"></a>
</h3>
<p>The <code>rse_tx</code> object contains the following assays:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assays</a></span><span class="op">(</span><span class="va">rse_tx</span><span class="op">)</span></span></code></pre></div>
<p><code>tpm</code>: normalized read counts of the 19,8093 genes across
900 samples.</p>
<p><code>Transcripts Per Million (TPM)</code> is a normalization method
for RNA-seq data that facilitates comparisons both within and between
samples. It adjusts for <code>gene length</code> and
<code>sequencing depth</code>, making it a helpful measure for relative
gene expression.</p>
<p>There are many ways to access the <code>tpm</code> data. Below some
of them:</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Extract the first element from the list of assays in the SummarizedExperiment object</span></span>
<span><span class="co"># This assumes that the first element in the list of assays is the one you are interested in</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assays</a></span><span class="op">(</span><span class="va">rse_tx</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://magrittr.tidyverse.org/reference/aliases.html" class="external-link">extract2</a></span><span class="op">(</span><span class="fl">1</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Another way of extracting the first element from the list of assays in the SummarizedExperiment object</span></span>
<span><span class="co"># Here, 'tpm' is assumed to be the name of the first assay</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assays</a></span><span class="op">(</span><span class="va">rse_tx</span><span class="op">)</span><span class="op">$</span><span class="va">tpm</span></span>
<span></span>
<span><span class="co"># Yet another way of extracting the first element from the list of assays in the SummarizedExperiment object</span></span>
<span><span class="co"># The simplify() function tries to simplify the result to a vector or matrix if possible</span></span>
<span><span class="co"># The first() function extracts the first element of the simplified result</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assays</a></span><span class="op">(</span><span class="va">rse_tx</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://purrr.tidyverse.org/reference/as_vector.html" class="external-link">simplify</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/nth.html" class="external-link">first</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="data-distribution">Data distribution<a class="anchor" aria-label="anchor" href="#data-distribution"></a>
</h3>
<p>Let’s have a look at the distribution of the tpm assay.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Transform 'tpm' data to a long format with 'Transcript', 'Sample', and 'TPM' columns</span></span>
<span><span class="va">tpm_tb</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log2</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assays</a></span><span class="op">(</span><span class="va">rse_tx</span><span class="op">)</span><span class="op">$</span><span class="va">tpm</span> <span class="op">+</span> <span class="fl">0.5</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html" class="external-link">rownames_to_column</a></span><span class="op">(</span>var <span class="op">=</span> <span class="st">"Transcript"</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://tidyr.tidyverse.org/reference/pivot_longer.html" class="external-link">pivot_longer</a></span><span class="op">(</span></span>
<span>        cols <span class="op">=</span> <span class="op">-</span><span class="va">Transcript</span>,</span>
<span>        names_to <span class="op">=</span> <span class="st">"Sample"</span>,</span>
<span>        values_to <span class="op">=</span> <span class="st">"TPM"</span></span>
<span>    <span class="op">)</span></span>
<span></span>
<span><span class="co"># Extract 'RNum', 'Region', and 'Dx' from 'rse_tx', convert to tibble</span></span>
<span><span class="va">colData_tb</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">rse_tx</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">RNum</span>, <span class="va">Region</span>, <span class="va">Dx</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://tibble.tidyverse.org/reference/as_tibble.html" class="external-link">as_tibble</a></span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Join 'tpm_tb' with 'colData_tb' based on 'Sample' and 'RNum'</span></span>
<span><span class="va">tpm_tb</span> <span class="op">&lt;-</span> <span class="va">tpm_tb</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate-joins.html" class="external-link">left_join</a></span><span class="op">(</span><span class="va">colData_tb</span>, by <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Sample"</span> <span class="op">=</span> <span class="st">"RNum"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Calculate mean TPM for each Dx and Region</span></span>
<span><span class="va">mean_tpm</span> <span class="op">&lt;-</span> <span class="va">tpm_tb</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html" class="external-link">group_by</a></span><span class="op">(</span><span class="va">Dx</span>, <span class="va">Region</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/summarise.html" class="external-link">summarise</a></span><span class="op">(</span></span>
<span>        mean_TPM <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/mean.html" class="external-link">mean</a></span><span class="op">(</span><span class="va">TPM</span>, na.rm <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>,</span>
<span>        .groups <span class="op">=</span> <span class="st">"drop"</span> <span class="co"># This drops the grouping</span></span>
<span>    <span class="op">)</span></span>
<span></span>
<span><span class="co">#  Dx      Region mean_TPM</span></span>
<span><span class="co">#   &lt;chr&gt;   &lt;chr&gt;     &lt;dbl&gt;</span></span>
<span><span class="co"># 1 Control DLPFC    0.0600</span></span>
<span><span class="co"># 2 Control HIPPO   -0.0385</span></span>
<span><span class="co"># 3 Schizo  DLPFC    0.0850</span></span>
<span><span class="co"># 4 Schizo  HIPPO   -0.0567</span></span>
<span></span>
<span><span class="co"># Create boxplot for first 100,000 rows, without outliers, split by Dx and Region</span></span>
<span><span class="va">tpm_tb</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/slice.html" class="external-link">slice</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">100000</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">Dx</span>, y <span class="op">=</span> <span class="va">TPM</span>, fill <span class="op">=</span> <span class="va">Dx</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_boxplot.html" class="external-link">geom_boxplot</a></span><span class="op">(</span>outlier.shape <span class="op">=</span> <span class="cn">NA</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html" class="external-link">facet_wrap</a></span><span class="op">(</span><span class="op">~</span><span class="va">Region</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_minimal</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span>title <span class="op">=</span> <span class="st">"Boxplot of TPM for Each Dx by Region"</span>, x <span class="op">=</span> <span class="st">"Dx"</span>, y <span class="op">=</span> <span class="st">"log2(TPM)+0.5"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Create a histogram of TPM for each Dx</span></span>
<span><span class="va">tpm_tb</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/slice.html" class="external-link">slice</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">100000</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">TPM</span>, fill <span class="op">=</span> <span class="va">Dx</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_histogram.html" class="external-link">geom_histogram</a></span><span class="op">(</span>alpha <span class="op">=</span> <span class="fl">0.5</span>, bins <span class="op">=</span> <span class="fl">30</span>, position <span class="op">=</span> <span class="st">"identity"</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html" class="external-link">facet_wrap</a></span><span class="op">(</span><span class="op">~</span><span class="va">Dx</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_minimal</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span>title <span class="op">=</span> <span class="st">"Histogram of TPM for Each Dx"</span>, x <span class="op">=</span> <span class="st">"log2(TPM)+0.5"</span>, y <span class="op">=</span> <span class="st">"Count"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Create a density plot of TPM for each Dx</span></span>
<span><span class="va">tpm_tb</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/slice.html" class="external-link">slice</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">100000</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">TPM</span>, fill <span class="op">=</span> <span class="va">Dx</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_density.html" class="external-link">geom_density</a></span><span class="op">(</span>alpha <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html" class="external-link">facet_wrap</a></span><span class="op">(</span><span class="op">~</span><span class="va">Dx</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_minimal</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span>title <span class="op">=</span> <span class="st">"Density Plot of TPM for Each Dx"</span>, x <span class="op">=</span> <span class="st">"log2(TPM)+0.5"</span>, y <span class="op">=</span> <span class="st">"Density"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="variance-partition-analysis">Variance Partition Analysis<a class="anchor" aria-label="anchor" href="#variance-partition-analysis"></a>
</h3>
<p>Evaluating the <code>correlation</code> among sample variables is an
imperative step in our analysis. The presence of
<code>highly correlated</code> variables can significantly impact the
<code>robustness</code> of our model, creating unstable estimates of
variance fractions. This instability is detrimental as it can obfuscate
the identification of the true contributors to expression variation.</p>
<style type="text/css">
<style>
p.comment {
background-color: #DBDBDB;
padding: 10px;
border: 1px solid black;
border-radius: 5px;
font-family: sans-serif;
}
</style>
<p class="comment">
</p>
<p>❓ <strong><em>Why assessing correlation is so
important?</em></strong> When two or more variables are highly
correlated, they essentially contain <code>redundant information</code>.
This redundancy can lead to what is known as
<code>multicollinearity</code>, a situation that can greatly
<code>destabilize</code> our model, potentially leading to misleading or
erroneous results.</p>
Therefore, assessing correlation enables us to
<code>prevent multicollinearity</code>, ensuring a more accurate and
reliable analysis.


<p>In essence, our objective is not just to recognize the factors
contributing to expression variation, but to accurately measure their
individual impacts. Understanding these relationships will allow us to
develop a more accurate and reliable predictive model.</p>
<p>Let’s first unlist RIN, mitoMapped and totalAssignedGene .</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">rse_tx</span><span class="op">)</span><span class="op">$</span><span class="va">RIN</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/numeric.html" class="external-link">as.numeric</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/unlist.html" class="external-link">unlist</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">rse_tx</span><span class="op">)</span><span class="op">$</span><span class="va">RIN</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">rse_tx</span><span class="op">)</span><span class="op">$</span><span class="va">mitoMapped</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/unlist.html" class="external-link">unlist</a></span><span class="op">(</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html" class="external-link">map</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">rse_tx</span><span class="op">)</span><span class="op">$</span><span class="va">mitoMapped</span>, <span class="fl">1</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">rse_tx</span><span class="op">)</span><span class="op">$</span><span class="va">totalAssignedGene</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/unlist.html" class="external-link">unlist</a></span><span class="op">(</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html" class="external-link">map</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">rse_tx</span><span class="op">)</span><span class="op">$</span><span class="va">totalAssignedGene</span>, <span class="fl">1</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">rse_tx</span><span class="op">)</span><span class="op">$</span><span class="va">rRNA_rate</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/unlist.html" class="external-link">unlist</a></span><span class="op">(</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html" class="external-link">map</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">rse_tx</span><span class="op">)</span><span class="op">$</span><span class="va">rRNA_rate</span>, <span class="fl">1</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">rse_tx</span><span class="op">)</span><span class="op">$</span><span class="va">numReads</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/unlist.html" class="external-link">unlist</a></span><span class="op">(</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html" class="external-link">map</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">rse_tx</span><span class="op">)</span><span class="op">$</span><span class="va">numReads</span>, <span class="fl">1</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">rse_tx</span><span class="op">)</span><span class="op">$</span><span class="va">numMapped</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/unlist.html" class="external-link">unlist</a></span><span class="op">(</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html" class="external-link">map</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">rse_tx</span><span class="op">)</span><span class="op">$</span><span class="va">numMapped</span>, <span class="fl">1</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">rse_tx</span><span class="op">)</span><span class="op">$</span><span class="va">numUnmapped</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/unlist.html" class="external-link">unlist</a></span><span class="op">(</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html" class="external-link">map</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">rse_tx</span><span class="op">)</span><span class="op">$</span><span class="va">numUnmapped</span>, <span class="fl">1</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">rse_tx</span><span class="op">)</span><span class="op">$</span><span class="va">overallMapRate</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/unlist.html" class="external-link">unlist</a></span><span class="op">(</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html" class="external-link">map</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">rse_tx</span><span class="op">)</span><span class="op">$</span><span class="va">overallMapRate</span>, <span class="fl">1</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">rse_tx</span><span class="op">)</span><span class="op">$</span><span class="va">concordMapRate</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/unlist.html" class="external-link">unlist</a></span><span class="op">(</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html" class="external-link">map</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">rse_tx</span><span class="op">)</span><span class="op">$</span><span class="va">concordMapRate</span>, <span class="fl">1</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">rse_tx</span><span class="op">)</span><span class="op">$</span><span class="va">totalMapped</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/unlist.html" class="external-link">unlist</a></span><span class="op">(</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html" class="external-link">map</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">rse_tx</span><span class="op">)</span><span class="op">$</span><span class="va">totalMapped</span>, <span class="fl">1</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">rse_tx</span><span class="op">)</span><span class="op">$</span><span class="va">mitoRate</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/unlist.html" class="external-link">unlist</a></span><span class="op">(</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html" class="external-link">map</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">rse_tx</span><span class="op">)</span><span class="op">$</span><span class="va">mitoRate</span>, <span class="fl">1</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">rse_tx</span><span class="op">)</span><span class="op">$</span><span class="va">ERCCsumLogErr</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/unlist.html" class="external-link">unlist</a></span><span class="op">(</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html" class="external-link">map</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">rse_tx</span><span class="op">)</span><span class="op">$</span><span class="va">ERCCsumLogErr</span>, <span class="fl">1</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p>We’ll be using the <code>canCorPairs</code> function from the
variancePartition which performs canonical correlation analysis (CCA)
between pairs of variables.</p>
<p>In the context of the <code>variancePartition</code> package, the
<code><a href="http://DiseaseNeurogenomics.github.io/variancePartition/reference/canCorPairs.html" class="external-link">canCorPairs()</a></code> computes the canonical correlations between
each pair of variables specified in the formula argument.</p>
<div class="alert alert-info">
<p><strong>Note:</strong> canCorPairs returns correlation values between
0 and 1. Canonical Correlation Analysis (CCA) returns sqrt(mean(rho^2)),
which is the fraction of the maximum possible correlation. When
comparing a two vectors, or a vector and a matrix, this gives the save
value as the absolute correlation. When comparing two sets of
categorical variables, this is equivalent to Cramer’s V statistic.</p>
</div>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">formula</span> <span class="op">&lt;-</span> <span class="op">~</span> <span class="va">Dx</span> <span class="op">+</span> <span class="va">RIN</span> <span class="op">+</span> <span class="va">Age</span> <span class="op">+</span> <span class="va">Sex</span> <span class="op">+</span> <span class="va">Race</span> <span class="op">+</span> <span class="va">mitoMapped</span> <span class="op">+</span> <span class="va">overallMapRate</span> <span class="op">+</span> <span class="va">mitoRate</span> <span class="op">+</span> <span class="va">rRNA_rate</span> <span class="op">+</span> <span class="va">concordMapRate</span> <span class="op">+</span> <span class="va">totalMapped</span> <span class="op">+</span> <span class="va">totalAssignedGene</span> <span class="op">+</span> <span class="va">Region</span></span>
<span><span class="va">correlation_matrix</span> <span class="op">&lt;-</span> <span class="fu"><a href="http://DiseaseNeurogenomics.github.io/variancePartition/reference/canCorPairs.html" class="external-link">canCorPairs</a></span><span class="op">(</span><span class="va">formula</span>, <span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">rse_tx</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Create the heatmap</span></span>
<span><span class="co"># Create a color function from green to purple</span></span>
<span><span class="va">color_func</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/colorRamp2/man/colorRamp2.html" class="external-link">colorRamp2</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="op">-</span><span class="fl">1</span>, <span class="fl">0</span>, <span class="fl">1</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"darkgreen"</span>, <span class="st">"white"</span>, <span class="st">"purple"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Create the heatmap</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/ComplexHeatmap/man/Heatmap.html" class="external-link">Heatmap</a></span><span class="op">(</span><span class="va">correlation_matrix</span>,</span>
<span>    name <span class="op">=</span> <span class="st">"Correlation"</span>, <span class="co"># Title for the color key</span></span>
<span>    show_row_names <span class="op">=</span> <span class="cn">TRUE</span>, <span class="co"># Show row names</span></span>
<span>    show_column_names <span class="op">=</span> <span class="cn">TRUE</span>, <span class="co"># Show column names</span></span>
<span>    cluster_rows <span class="op">=</span> <span class="cn">FALSE</span>, <span class="co"># Do not cluster rows</span></span>
<span>    cluster_columns <span class="op">=</span> <span class="cn">FALSE</span>, <span class="co"># Do not cluster columns</span></span>
<span>    cell_fun <span class="op">=</span> <span class="kw">function</span><span class="op">(</span><span class="va">j</span>, <span class="va">i</span>, <span class="va">x</span>, <span class="va">y</span>, <span class="va">width</span>, <span class="va">height</span>, <span class="va">fill</span><span class="op">)</span> <span class="op">{</span></span>
<span>        <span class="fu"><a href="https://rdrr.io/r/grid/grid.text.html" class="external-link">grid.text</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/sprintf.html" class="external-link">sprintf</a></span><span class="op">(</span><span class="st">"%.2f"</span>, <span class="va">correlation_matrix</span><span class="op">[</span><span class="va">i</span>, <span class="va">j</span><span class="op">]</span><span class="op">)</span>, <span class="va">x</span>, <span class="va">y</span>,</span>
<span>            gp <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/grid/gpar.html" class="external-link">gpar</a></span><span class="op">(</span>fontsize <span class="op">=</span> <span class="fl">10</span><span class="op">)</span></span>
<span>        <span class="op">)</span></span>
<span>    <span class="op">}</span>,</span>
<span>    col <span class="op">=</span> <span class="va">color_func</span>, <span class="co"># Use the custom color function</span></span>
<span>    heatmap_legend_param <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span></span>
<span>        title <span class="op">=</span> <span class="st">"Correlation"</span>, <span class="co"># legend title</span></span>
<span>        color_bar <span class="op">=</span> <span class="st">"continuous"</span>, <span class="co"># type of color bar</span></span>
<span>        legend_direction <span class="op">=</span> <span class="st">"horizontal"</span> <span class="co"># direction of legend</span></span>
<span>    <span class="op">)</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p>The <code>cell_fun</code> function customizes the display of each
cell in the heatmap by adding the correlation value as text inside the
cell with a specific format and font size.</p>
<p>From the heatmap we can see that there’s a high correlation between
<code>mitoMapped</code> and <code>totalAssignedGene</code>.</p>
<div class="alert alert-info">
<p><strong>Note:</strong> One of the strategies to address
multicollinearity is removing some of the highly correlated independent
variables.</p>
</div>
</div>
</div>
<div class="section level2">
<h2 id="qsvar-quality-surrogate-variable-analysis">qsvaR: Quality Surrogate Variable Analysis<a class="anchor" aria-label="anchor" href="#qsvar-quality-surrogate-variable-analysis"></a>
</h2>
<p>Here’s a general outline of how qSVA can be used to adjust for
degradation in postmortem brain tissue:</p>
<p><strong>Identify Surrogate Variables</strong>: The first step in SVA
is to identify surrogate variables that capture the unwanted variation
in the data. These variables are often not of primary interest in the
study but can significantly affect the results.</p>
<p><strong>Estimate the Effect of Surrogate Variables</strong>: The next
step is to estimate the effect of the surrogate variables on the data.
This is typically done using a statistical model that includes the
surrogate variables as covariates. The coefficients for the surrogate
variables in this model represent their effect on the data.</p>
<p><strong>Adjust the Data for the Effect of Surrogate
Variables</strong>: Once the effect of the surrogate variables has been
estimated, this effect can be removed from the data. This is typically
done by subtracting the estimated effect of the surrogate variables from
the data. The resulting adjusted data should be free of the unwanted
variation due to degradation.</p>
<p><strong>Perform the Primary Analysis on the Adjusted Data</strong>:
Once the data has been adjusted for the effect of the surrogate
variables, the primary analysis can be performed on the adjusted data.
This analysis should yield results that are not confounded by the
unwanted variation due to degradation.</p>
<div class="section level3">
<h3 id="degradation-transcripts">Degradation transcripts<a class="anchor" aria-label="anchor" href="#degradation-transcripts"></a>
</h3>
<p>After loading the <code>rse_tx</code> object we subset for the
transcripts associated with degradation.</p>
<p>In qsvaR there are three models to choose from. Here the names
<code>cell_component</code>, <code>top1500</code>, and
<code>standard</code> refer to models that were determined to be
effective in removing degradation effects.</p>
</div>
<div class="section level3">
<h3 id="qsvar-models">qsvaR models<a class="anchor" aria-label="anchor" href="#qsvar-models"></a>
</h3>
<div class="section level4">
<h4 id="the-standard-model">The standard model<a class="anchor" aria-label="anchor" href="#the-standard-model"></a>
</h4>
<p>The <code>standard</code> model involves taking the union of the
<code>top 1000</code> transcripts associated with degradation from the
interaction model and the main effect model.</p>
</div>
<div class="section level4">
<h4 id="the-top1500-model">The top1500 model<a class="anchor" aria-label="anchor" href="#the-top1500-model"></a>
</h4>
<p>The <code>top1500</code> model is the same as the “standard” model
except the union of the <code>top 1500</code> genes associated with
degradation is selected.</p>
</div>
<div class="section level4">
<h4 id="the-cell-component-model">The cell component model<a class="anchor" aria-label="anchor" href="#the-cell-component-model"></a>
</h4>
<p>The <code>cell_component</code> model, involved
<code>deconvolution</code> of the degradation matrix to determine the
proportion of cell types within our studied tissue. These proportions
were then added to our <code><a href="https://rdrr.io/r/stats/model.matrix.html" class="external-link">model.matrix()</a></code> and the union of the
top 1000 transcripts in the interaction model, the main effect model,
and the cell proportions model were used to generate this model of qSVs.
It’s the most effective of our models.</p>
</div>
</div>
<div class="section level3">
<h3 id="run-qsva">Run qSVA<a class="anchor" aria-label="anchor" href="#run-qsva"></a>
</h3>
<p>In this example we will choose <code>cell_component</code> when using
the <code><a href="https://rdrr.io/pkg/qsvaR/man/getDegTx.html" class="external-link">getDegTx()</a></code> and <code><a href="https://rdrr.io/pkg/qsvaR/man/select_transcripts.html" class="external-link">select_transcripts()</a></code>
functions. <code><a href="https://rdrr.io/pkg/qsvaR/man/getDegTx.html" class="external-link">getDegTx()</a></code> is used to select a specific set of
transcripts from a given experiment, and provides a warning if the mean
expression level of these transcripts is low.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Next we get the degraded transcripts for qSVA from the "cell_component"</span></span>
<span><span class="co">## model</span></span>
<span><span class="va">DegTx</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/qsvaR/man/getDegTx.html" class="external-link">getDegTx</a></span><span class="op">(</span><span class="va">rse_tx</span>, type <span class="op">=</span> <span class="st">"cell_component"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">DegTx</span><span class="op">)</span></span>
<span><span class="co"># [1] 2976  900</span></span></code></pre></div>
<p>The <code><a href="https://rdrr.io/pkg/qsvaR/man/getPCs.html" class="external-link">getPCs()</a></code> applies PCA using the <code>prcomp</code>
function.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Now we can compute the Principal Components (PCs) of the degraded transcripts</span></span>
<span><span class="va">pcTx</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/qsvaR/man/getPCs.html" class="external-link">getPCs</a></span><span class="op">(</span><span class="va">DegTx</span>, <span class="st">"tpm"</span><span class="op">)</span></span></code></pre></div>
<p>We use a model matrix that accounts for relevant variables. We
include variables such as Age, Sex, Race, and Region in this model. Our
dataset, DegTx, a RangedSummarizedExperiment object, is used as the
rse_tx input.</p>
<div class="alert alert-info">
<p><strong>Note:</strong> The model.matrix() function takes a formula
and a data frame (or similar object) and returns the design matrix.</p>
<p>model.matrix() will automatically create dummy variables for each
level of the factor. Dummy variables are binary (0/1) variables that
indicate the presence of a categorical level for each observation.</p>
</div>
<p>The <code><a href="https://rdrr.io/r/stats/model.matrix.html" class="external-link">model.matrix()</a></code> takes the variables in the formula
format:</p>
<p><code>model.matrix([target] ~ [predictor / features], data = [data source])</code></p>
<p>This part is called a formula:
<code>~ Dx + Age + Sex + Race + Region</code></p>
<p>The tilde (~) can be read as <code>is modeled as</code>. The
variables on the right side of the tilde are the predictor variables (or
features).</p>
<p>In this case, Dx, Age, Sex, Race, and Region are your
<code>predictors</code>.</p>
<p>data = colData(rse_tx): This argument tells R which dataset to use
for creating the design matrix.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Creates a model matrix for regression analysis using the predictors Dx, Age, Sex, Race, and Region from the column data of the rse_tx object.</span></span>
<span><span class="va">mod</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/model.matrix.html" class="external-link">model.matrix</a></span><span class="op">(</span><span class="op">~</span> <span class="va">Dx</span> <span class="op">+</span> <span class="va">Age</span> <span class="op">+</span> <span class="va">Sex</span> <span class="op">+</span> <span class="va">Race</span> <span class="op">+</span> <span class="va">Region</span>,</span>
<span>    data <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">rse_tx</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># check the names of the columns in the mod matrix</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">mod</span><span class="op">)</span></span>
<span><span class="co"># [1] "(Intercept)" "DxSchizo"    "Age"         "SexM"        "RaceAS"</span></span>
<span><span class="co"># [6] "RaceCAUC"    "RaceHISP"    "RegionHIPPO"</span></span></code></pre></div>
<p>The <code><a href="https://rdrr.io/pkg/qsvaR/man/k_qsvs.html" class="external-link">k_qsvs()</a></code> function is employed to determine the
required number of Principal Components (PCs) needed to capture the
underlying variation in our data.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Get the number of qsvs</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">20230703</span><span class="op">)</span></span>
<span><span class="va">k</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/qsvaR/man/k_qsvs.html" class="external-link">k_qsvs</a></span><span class="op">(</span><span class="va">DegTx</span>, <span class="va">mod</span>, <span class="st">"tpm"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">k</span><span class="op">)</span></span>
<span><span class="co"># [1] 34</span></span></code></pre></div>
<p>You can simplify the process by utilizing our integrated function,
<code>qSVA</code>. This wrapper function conveniently combines all
previously mentioned functions, streamlining the analysis into a single
step.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Example use of the wrapper function qSVA()</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">20230703</span><span class="op">)</span></span>
<span><span class="va">qsva_pcs_cc</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/qsvaR/man/qSVA.html" class="external-link">qSVA</a></span><span class="op">(</span><span class="va">rse_tx</span>, type <span class="op">=</span> <span class="st">"cell_component"</span>, mod <span class="op">=</span> <span class="va">mod</span>, assayname <span class="op">=</span> <span class="st">"tpm"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">qsva_pcs_cc</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 900  34</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="differential-expression-using-limma">Differential Expression using limma<a class="anchor" aria-label="anchor" href="#differential-expression-using-limma"></a>
</h3>
<p><code>limma</code> fits a so-called linear model. The limma User’s
Guide from the limma webpage.</p>
<p>In the context of limma, each gene’s expression level is modeled as a
linear combination of the experimental factors, with some added noise or
error term.</p>
<p>For example, consider a simple model for one gene:</p>
<p><span class="math inline">\(expression = intercept + beta*treatment +
error\)</span></p>
<p>Here:</p>
<p><strong>expression</strong> is the expression level of the gene.</p>
<p><strong>intercept</strong> is the base level of gene expression
(i.e., in the control group).</p>
<p><strong>beta</strong> is a coefficient representing the effect size
of the treatment.</p>
<p><strong>treatment</strong> is a variable representing whether the
sample is a control or treatment (often coded as 0 for control, 1 for
treatment).</p>
<p><strong>error</strong> is the noise or unexplained variability in
gene expression.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Add the qSVs to our statistical model</span></span>
<span><span class="va">mod_qSVA</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/cbind.html" class="external-link">cbind</a></span><span class="op">(</span></span>
<span>    <span class="va">mod</span>,</span>
<span>    <span class="va">qsva_pcs_cc</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><code>mod_qSVA</code> includes both our original model and the qSVs.
The qSVs adjust for the degradation effect.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Extract the transcript expression values and put them in the</span></span>
<span><span class="co">## log2(TPM + 1) scale</span></span>
<span><span class="va">txExprs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log2</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assays</a></span><span class="op">(</span><span class="va">rse_tx</span><span class="op">)</span><span class="op">$</span><span class="va">tpm</span> <span class="op">+</span> <span class="fl">0.5</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Run the standard linear model for differential expression</span></span>
<span><span class="va">fitTx</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/limma/man/lmFit.html" class="external-link">lmFit</a></span><span class="op">(</span><span class="va">txExprs</span>, <span class="va">mod_qSVA</span><span class="op">)</span></span>
<span><span class="va">eBTx</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/limma/man/ebayes.html" class="external-link">eBayes</a></span><span class="op">(</span><span class="va">fitTx</span><span class="op">)</span></span></code></pre></div>
<p>In this part above of the code, we are running a linear model for
differential expression analysis using the <code><a href="https://rdrr.io/pkg/limma/man/lmFit.html" class="external-link">lmFit()</a></code> from the
limma package in R.</p>
<p>The function takes two main arguments: the matrix of expression
values (txExprs) and the design matrix (mod_qSVA).</p>
<p>The design matrix specifies the linear model that you’re fitting to
the data. In this case, mod_qSVA likely includes the variables of
interest and the qSVs.</p>
<p>The <code><a href="https://rdrr.io/pkg/limma/man/lmFit.html" class="external-link">lmFit()</a></code> function fits a separate linear model to
each gene (or transcript, in this case), and the result (fitTx) is an
object that contains the fitted model coefficients and other statistics
for each gene.</p>
<p>After fitting the model, <code><a href="https://rdrr.io/pkg/limma/man/ebayes.html" class="external-link">eBayes()</a></code> is used to compute
<code>empirical Bayes statistics</code> for differential expression. The
function applies a shrinkage technique to the standard errors, improving
the stability of the estimates, particularly when the number of samples
is small.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">eBTx</span><span class="op">)</span></span>
<span><span class="co">#  [1] "(Intercept)" "DxSchizo"    "Age"         "SexM"        "RaceAS"</span></span>
<span><span class="co">#  [6] "RaceCAUC"    "RaceHISP"    "RegionHIPPO" "qSV1"        "qSV2"</span></span>
<span><span class="co"># [11] "qSV3"        "qSV4"        "qSV5"        "qSV6"        "qSV7"</span></span>
<span><span class="co"># [16] "qSV8"        "qSV9"        "qSV10"       "qSV11"       "qSV12"</span></span>
<span><span class="co"># [21] "qSV13"       "qSV14"       "qSV15"       "qSV16"       "qSV17"</span></span>
<span><span class="co"># [26] "qSV18"       "qSV19"       "qSV20"       "qSV21"       "qSV22"</span></span>
<span><span class="co"># [31] "qSV23"       "qSV24"       "qSV25"       "qSV26"       "qSV27"</span></span>
<span><span class="co"># [36] "qSV28"       "qSV29"       "qSV30"       "qSV31"       "qSV32"</span></span>
<span><span class="co"># [41] "qSV33"       "qSV34"</span></span></code></pre></div>
<p>Let’s do the differential expression analysis.</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Extract the differential expression results</span></span>
<span><span class="va">sigTx</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/limma/man/toptable.html" class="external-link">topTable</a></span><span class="op">(</span><span class="va">eBTx</span>,</span>
<span>    coef <span class="op">=</span> <span class="st">"DxSchizo"</span>,</span>
<span>    p.value <span class="op">=</span> <span class="fl">1</span>, number <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">rse_tx</span><span class="op">)</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><code>coef = "DxSchizo"</code> specifies the model coefficient of
interest. The <code><a href="https://rdrr.io/pkg/limma/man/toptable.html" class="external-link">topTable()</a></code> will return results for this
specific term.</p>
<p>Let’s explore the top results.</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Explore the top results</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">sigTx</span><span class="op">)</span></span></code></pre></div>
<p>Here’s a breakdown of what each column in this output represents:</p>
<p><strong>logFC (Log Fold Change)</strong>: This represents the log2
fold-change for the differential expression between two conditions. A
positive value indicates higher expression in the second condition,
while a negative value indicates higher expression in the first
condition. For example, a logFC of -0.07856532 for ENST00000553142.5
means this transcript has lower expression in the second condition
compared to the first.</p>
<p><strong>AveExpr (Average Expression)</strong>: This is the average
log2 expression level of the gene across all samples. It provides a
general idea of how abundantly the gene is expressed.</p>
<p><strong>t (t-statistic)</strong>: This is the t-statistic from the
differential expression test. It is a measure of how different the
expression levels of the gene are between the two conditions, in units
of standard error.</p>
<div class="alert alert-info">
<p><strong>Note:</strong> A higher absolute value of the t-statistic
means the gene’s expression is more significantly different between the
conditions.</p>
</div>
<p><strong>P.Value (p-value)</strong>: This is the p-value from the
differential expression test. It quantifies the statistical evidence
against the null hypothesis that the gene’s expression levels are the
same in the two conditions.</p>
<div class="alert alert-info">
<p><strong>Note:</strong> A lower p-value suggests stronger evidence
against the null hypothesis.</p>
</div>
<p><strong>adj.P.Val (Adjusted p-value)</strong>: This is the p-value
adjusted for multiple testing using the method of Benjamini and
Hochberg, also known as the false discovery rate (FDR). It helps control
for false positives when testing multiple hypotheses (genes) at
once.</p>
<div class="alert alert-info">
<p><strong>Note:</strong> Typically, genes with an adjusted p-value (or
FDR) less than 0.05 or 0.01 are considered statistically significantly
differentially expressed.</p>
</div>
<p><strong>B (B-statistic or log-odds of differential
expression)</strong>: This is the B-statistic from the empirical Bayes
analysis, which can be interpreted as the log-odds that the gene is
differentially expressed.</p>
<div class="alert alert-info">
<p><strong>Note:</strong> A higher B-statistic means the gene is more
likely to be differentially expressed.</p>
</div>
</div>
<div class="section level3">
<h3 id="differential-expression-quality-or-dequal-plots">Differential expression quality or DEqual plots<a class="anchor" aria-label="anchor" href="#differential-expression-quality-or-dequal-plots"></a>
</h3>
<p>This R function, <code>DEqual</code>, is designed to compare
degradation t-statistics with the t-statistics from a differential
expression (DE) analysis.</p>
<p>It takes as input a data frame (DE) containing differential
expression results, such as those outputted by the limma package’s
topTable function.</p>
<p>The function identifies the transcripts that are common between the
<code>DE dataframe</code> and <code>degradation_tstats</code> from the
qsvaR package and creates a new dataframe, common_data, which contains
the <code>degradation t-statistics</code> and
<code>DE t-statistics</code> for the common transcripts.</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Generate a DEqual() plot using the model results with qSVs</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/qsvaR/man/DEqual.html" class="external-link">DEqual</a></span><span class="op">(</span><span class="va">sigTx</span><span class="op">)</span></span></code></pre></div>
<p>The color of each bin represents the number of data points (in this
case, transcripts) within that bin. This is set by the
<code><a href="https://ggplot2.tidyverse.org/reference/scale_colour_continuous.html" class="external-link">scale_fill_continuous()</a></code>, which maps the continuous number
of data points in each bin to a continuous color scale.</p>
<p>The <code>type = "viridis"</code> argument specifies the use of the
viridis color scale. The viridis color scale ranges from dark purple
(for low values) through blue, green, yellow, and up to bright yellow
(for high values).</p>
<p>In this plot:</p>
<p><strong>Dark purple to blue bins</strong>: These colors represent
bins with a low number of data points. If a bin is dark purple or blue,
it means there are relatively few transcripts with those particular DE
and degradation t-statistics.</p>
<p><strong>Green to yellow bins</strong>: These colors represent bins
with a moderate number of data points. If a bin is green or yellow, it
means there are a moderate number of transcripts with those particular
DE and degradation t-statistics.</p>
<p><strong>Bright yellow bins</strong>: These colors represent bins with
a high number of data points. If a bin is bright yellow, it means there
are many transcripts with those particular DE and degradation
t-statistics.</p>
<p>For comparison, here is the DEqual() plot for the model without
qSVs.</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Generate a DEqual() plot using the model results without qSVs</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/qsvaR/man/DEqual.html" class="external-link">DEqual</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/limma/man/toptable.html" class="external-link">topTable</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/limma/man/ebayes.html" class="external-link">eBayes</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/limma/man/lmFit.html" class="external-link">lmFit</a></span><span class="op">(</span><span class="va">txExprs</span>, <span class="va">mod</span><span class="op">)</span><span class="op">)</span>, coef <span class="op">=</span> <span class="st">"DxSchizo"</span>, p.value <span class="op">=</span> <span class="fl">1</span>, number <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">rse_tx</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p>The results from the differential expression analysis without qSVA
normalization are shown in the two DEqual plots. The contrast between
the two plots is evident and reveals the effectiveness of degradation
effect removal.</p>
<p>In the first plot, the correlation stands at -0.038, suggesting a
minimal relationship between degradation and the differential expression
results. This indicates that the degradation effects have been
effectively eliminated from the data, leading to a cleaner
interpretation of the gene expression differences.</p>
<p>Conversely, the second plot reveals a correlation of 0.5 with the
degradation experiment, even after adjusting for several common
variables. Such a high correlation underscores the fact that
degradation’s signal remains significantly intertwined with our data.
This can potentially introduce confounding factors in our differential
expression analysis between the schizophrenia (SCZD) and neurotypical
control groups.</p>
</div>
</div>
<div class="section level2">
<h2 id="conclusion">Conclusion<a class="anchor" aria-label="anchor" href="#conclusion"></a>
</h2>
<p>In essence, the high correlation in the second plot could be masking
the true differential gene expression between the case-control groups by
mingling the degradation signal into the expression data. This
emphasizes the need for adequate normalization, such as qSVA, to reduce
such confounding factors and ensure a more accurate representation of
the biological differences in gene expression.</p>
</div>
<div class="section level2">
<h2 id="reproducibility">Reproducibility<a class="anchor" aria-label="anchor" href="#reproducibility"></a>
</h2>
<p>The <em><a href="https://bioconductor.org/packages/3.15/qsvaRPkgDemo" class="external-link">qsvaRPkgDemo</a></em>
package (HediaTnani, 2023) was made possible thanks to:</p>
<ul>
<li>R (R Core Team, 2022)</li>
<li>
<em><a href="https://bioconductor.org/packages/3.15/BiocStyle" class="external-link">BiocStyle</a></em>
(Oleś, 2022)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=knitr" class="external-link">knitr</a></em> (Xie,
2023)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=RefManageR" class="external-link">RefManageR</a></em>
(McLean, 2017)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=rmarkdown" class="external-link">rmarkdown</a></em>
(Allaire, Xie, Dervieux, McPherson, Luraschi, Ushey, Atkins, Wickham,
Cheng, Chang, and Iannone, 2023)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=sessioninfo" class="external-link">sessioninfo</a></em>
(Wickham, Chang, Flight, Müller, and Hester, 2021)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=testthat" class="external-link">testthat</a></em>
(Wickham, 2011)</li>
</ul>
<p>This package was developed using <em><a href="https://bioconductor.org/packages/3.15/biocthis" class="external-link">biocthis</a></em>.</p>
<p>Code for creating the vignette</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Create the vignette</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/rstudio/rmarkdown" class="external-link">"rmarkdown"</a></span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/system.time.html" class="external-link">system.time</a></span><span class="op">(</span><span class="fu"><a href="https://pkgs.rstudio.com/rmarkdown/reference/render.html" class="external-link">render</a></span><span class="op">(</span><span class="st">"qsvaRPkgDemo.Rmd"</span>, <span class="st">"BiocStyle::html_document"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Extract the R code</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://yihui.org/knitr/" class="external-link">"knitr"</a></span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/knit.html" class="external-link">knit</a></span><span class="op">(</span><span class="st">"qsvaRPkgDemo.Rmd"</span>, tangle <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<p>Date the vignette was generated.</p>
<pre><code><span><span class="co">#&gt; [1] "2023-07-25 08:18:10 EDT"</span></span></code></pre>
<p>Wallclock time spent generating the vignette.</p>
<pre><code><span><span class="co">#&gt; Time difference of 2.036 secs</span></span></code></pre>
<p><code>R</code> session information.</p>
<pre><code><span><span class="co">#&gt; <span style="color: #00BBBB; font-weight: bold;">─ Session info ───────────────────────────────────────────────────────────────────────────────────────────────────────</span></span></span>
<span><span class="co">#&gt;  <span style="color: #555555; font-style: italic;">setting </span> <span style="color: #555555; font-style: italic;">value</span></span></span>
<span><span class="co">#&gt;  version  R version 4.2.1 (2022-06-23)</span></span>
<span><span class="co">#&gt;  os       macOS Big Sur ... 10.16</span></span>
<span><span class="co">#&gt;  system   x86_64, darwin17.0</span></span>
<span><span class="co">#&gt;  ui       X11</span></span>
<span><span class="co">#&gt;  language en</span></span>
<span><span class="co">#&gt;  collate  en_US.UTF-8</span></span>
<span><span class="co">#&gt;  ctype    en_US.UTF-8</span></span>
<span><span class="co">#&gt;  tz       America/New_York</span></span>
<span><span class="co">#&gt;  date     2023-07-25</span></span>
<span><span class="co">#&gt;  pandoc   3.1.1 @ /Applications/RStudio.app/Contents/Resources/app/quarto/bin/tools/ (via rmarkdown)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB; font-weight: bold;">─ Packages ───────────────────────────────────────────────────────────────────────────────────────────────────────────</span></span></span>
<span><span class="co">#&gt;  <span style="color: #555555; font-style: italic;">package    </span> <span style="color: #555555; font-style: italic;">*</span> <span style="color: #555555; font-style: italic;">version  </span> <span style="color: #555555; font-style: italic;">date (UTC)</span> <span style="color: #555555; font-style: italic;">lib</span> <span style="color: #555555; font-style: italic;">source</span></span></span>
<span><span class="co">#&gt;  backports     1.4.1     <span style="color: #555555;">2021-12-13</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  bibtex        0.5.1     <span style="color: #555555;">2023-01-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  BiocManager   1.30.21.1 <span style="color: #555555;">2023-07-18</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  BiocStyle   * 2.24.0    <span style="color: #555555;">2022-04-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  bookdown      0.34      <span style="color: #555555;">2023-05-09</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  bslib         0.5.0     <span style="color: #555555;">2023-06-09</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  cachem        1.0.8     <span style="color: #555555;">2023-05-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  cli           3.6.1     <span style="color: #555555;">2023-03-23</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.1)</span></span></span>
<span><span class="co">#&gt;  desc          1.4.2     <span style="color: #555555;">2022-09-08</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  digest        0.6.33    <span style="color: #555555;">2023-07-07</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  evaluate      0.21      <span style="color: #555555;">2023-05-05</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  fastmap       1.1.1     <span style="color: #555555;">2023-02-24</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  fs            1.6.3     <span style="color: #555555;">2023-07-20</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  generics      0.1.3     <span style="color: #555555;">2022-07-05</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  glue          1.6.2     <span style="color: #555555;">2022-02-24</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  htmltools     0.5.5     <span style="color: #555555;">2023-03-23</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.1)</span></span></span>
<span><span class="co">#&gt;  httr          1.4.6     <span style="color: #555555;">2023-05-08</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  jquerylib     0.1.4     <span style="color: #555555;">2021-04-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  jsonlite      1.8.7     <span style="color: #555555;">2023-06-29</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  knitr         1.43      <span style="color: #555555;">2023-05-25</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  lifecycle     1.0.3     <span style="color: #555555;">2022-10-07</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  lubridate     1.9.2     <span style="color: #555555;">2023-02-10</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  magrittr      2.0.3     <span style="color: #555555;">2022-03-30</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  memoise       2.0.1     <span style="color: #555555;">2021-11-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  pkgdown       2.0.7     <span style="color: #555555;">2022-12-14</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  plyr          1.8.8     <span style="color: #555555;">2022-11-11</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  purrr         1.0.1     <span style="color: #555555;">2023-01-10</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  R6            2.5.1     <span style="color: #555555;">2021-08-19</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  ragg          1.2.5     <span style="color: #555555;">2023-01-12</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  Rcpp          1.0.11    <span style="color: #555555;">2023-07-06</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  RefManageR  * 1.4.0     <span style="color: #555555;">2022-09-30</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  rlang         1.1.1     <span style="color: #555555;">2023-04-28</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  rmarkdown     2.23      <span style="color: #555555;">2023-07-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  rprojroot     2.0.3     <span style="color: #555555;">2022-04-02</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  rstudioapi    0.15.0    <span style="color: #555555;">2023-07-07</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  sass          0.4.7     <span style="color: #555555;">2023-07-15</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  sessioninfo * 1.2.2     <span style="color: #555555;">2021-12-06</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  stringi       1.7.12    <span style="color: #555555;">2023-01-11</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  stringr       1.5.0     <span style="color: #555555;">2022-12-02</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  systemfonts   1.0.4     <span style="color: #555555;">2022-02-11</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  textshaping   0.3.6     <span style="color: #555555;">2021-10-13</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  timechange    0.2.0     <span style="color: #555555;">2023-01-11</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  vctrs         0.6.3     <span style="color: #555555;">2023-06-14</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  xfun          0.39      <span style="color: #555555;">2023-04-20</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  xml2          1.3.5     <span style="color: #555555;">2023-07-06</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  yaml          2.3.7     <span style="color: #555555;">2023-01-23</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #555555;"> [1] /Library/Frameworks/R.framework/Versions/4.2/Resources/library</span></span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB; font-weight: bold;">──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────</span></span></span></code></pre>
</div>
<div class="section level2">
<h2 id="bibliography">Bibliography<a class="anchor" aria-label="anchor" href="#bibliography"></a>
</h2>
<p>This vignette was generated using <em><a href="https://bioconductor.org/packages/3.15/BiocStyle" class="external-link">BiocStyle</a></em>
(Oleś, 2022) with <em><a href="https://CRAN.R-project.org/package=knitr" class="external-link">knitr</a></em> (Xie,
2023) and <em><a href="https://CRAN.R-project.org/package=rmarkdown" class="external-link">rmarkdown</a></em>
(Allaire, Xie, Dervieux et al., 2023) running behind the scenes.</p>
<p>Citations made with <em><a href="https://CRAN.R-project.org/package=RefManageR" class="external-link">RefManageR</a></em>
(McLean, 2017).</p>
<p>
<a id="bib-allaire2023rmarkdown"></a><a href="#cite-allaire2023rmarkdown">[1]</a><cite>
J. Allaire, Y. Xie, C. Dervieux, et al. <em>rmarkdown: Dynamic Documents
for R</em>. R package version 2.23. 2023. URL:
<a href="https://github.com/rstudio/rmarkdown" class="external-link">https://github.com/rstudio/rmarkdown</a>.</cite>
</p>
<p>
<a id="bib-hediatnani2023differential"></a><a href="#cite-hediatnani2023differential">[2]</a><cite>
HediaTnani. <em>Differential Expression Analysis using Limma and
qsvaR</em>. <a href="https://github.com/HediaTnani/qsvaRPkgDemo/qsvaRPkgDemo" class="external-link uri">https://github.com/HediaTnani/qsvaRPkgDemo/qsvaRPkgDemo</a>
- R package version 0.99.0. 2023. DOI:
<a href="https://doi.org/10.18129/B9.bioc.qsvaRPkgDemo" class="external-link">10.18129/B9.bioc.qsvaRPkgDemo</a>.
URL:
<a href="http://www.bioconductor.org/packages/qsvaRPkgDemo" class="external-link">http://www.bioconductor.org/packages/qsvaRPkgDemo</a>.</cite>
</p>
<p>
<a id="bib-mclean2017refmanager"></a><a href="#cite-mclean2017refmanager">[3]</a><cite>
M. W. McLean. “RefManageR: Import and Manage BibTeX and BibLaTeX
References in R”. In: <em>The Journal of Open Source Software</em>
(2017). DOI:
<a href="https://doi.org/10.21105/joss.00338" class="external-link">10.21105/joss.00338</a>.</cite>
</p>
<p>
<a id="bib-ole2022biocstyle"></a><a href="#cite-ole2022biocstyle">[4]</a><cite>
A. Oleś. <em>BiocStyle: Standard styles for vignettes and other
Bioconductor documents</em>. R package version 2.24.0. 2022. URL:
<a href="https://github.com/Bioconductor/BiocStyle" class="external-link">https://github.com/Bioconductor/BiocStyle</a>.</cite>
</p>
<p>
<a id="bib-2022language"></a><a href="#cite-2022language">[5]</a><cite>
R Core Team. <em>R: A Language and Environment for Statistical
Computing</em>. R Foundation for Statistical Computing. Vienna, Austria,
2022. URL:
<a href="https://www.R-project.org/" class="external-link">https://www.R-project.org/</a>.</cite>
</p>
<p>
<a id="bib-wickham2011testthat"></a><a href="#cite-wickham2011testthat">[6]</a><cite>
H. Wickham. “testthat: Get Started with Testing”. In: <em>The R
Journal</em> 3 (2011), pp. 5–10. URL:
<a href="https://journal.r-project.org/archive/2011-1/RJournal_2011-1_Wickham.pdf" class="external-link">https://journal.r-project.org/archive/2011-1/RJournal_2011-1_Wickham.pdf</a>.</cite>
</p>
<p>
<a id="bib-wickham2021sessioninfo"></a><a href="#cite-wickham2021sessioninfo">[7]</a><cite>
H. Wickham, W. Chang, R. Flight, et al. <em>sessioninfo: R Session
Information</em>. R package version 1.2.2. 2021. URL:
<a href="https://CRAN.R-project.org/package=sessioninfo" class="external-link">https://CRAN.R-project.org/package=sessioninfo</a>.</cite>
</p>
<p>
<a id="bib-xie2023knitr"></a><a href="#cite-xie2023knitr">[8]</a><cite>
Y. Xie. <em>knitr: A General-Purpose Package for Dynamic Report
Generation in R</em>. R package version 1.43. 2023. URL:
<a href="https://yihui.org/knitr/" class="external-link">https://yihui.org/knitr/</a>.</cite>
</p>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Hedia Tnani.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
