<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Differential Expression Analysis using Limma and qsvaR • qsvaRPkgDemo</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet">
<meta property="og:title" content="Differential Expression Analysis using Limma and qsvaR">
<meta property="og:description" content="qsvaRPkgDemo">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">qsvaRPkgDemo</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.99.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/qsvaRPkgDemo.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/LieberInstitute/qsvaRPkgDemo/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Differential Expression Analysis using Limma and
qsvaR</h1>
                        <h4 data-toc-skip class="author">Hedia
Tnani</h4>
            <address class="author_afil">
      Lieber Institute for Brain Development, Johns Hopkins Medical
Campus<br><a class="author_email" href="mailto:#"></a><a href="mailto:hediatnani0@gmail.com" class="email">hediatnani0@gmail.com</a>
      </address>
                              <h4 data-toc-skip class="author">Joshua M.
Stolz</h4>
            <address class="author_afil">
      Lieber Institute for Brain Development, Johns Hopkins Medical
Campus<br><a class="author_email" href="mailto:#"></a><a href="mailto:jstolz80@gmail.com" class="email">jstolz80@gmail.com</a>
      </address>
                              <h4 data-toc-skip class="author">Louise A.
Huuki-Myers</h4>
            <address class="author_afil">
      Lieber Institute for Brain Development, Johns Hopkins Medical
Campus<br><a class="author_email" href="mailto:#"></a><a href="mailto:lahuuki@gmail.com" class="email">lahuuki@gmail.com</a>
      </address>
                              <h4 data-toc-skip class="author">Leonardo
Collado-Torres</h4>
            <address class="author_afil">
      Lieber Institute for Brain Development, Johns Hopkins Medical
Campus<br><a class="author_email" href="mailto:#"></a><a href="mailto:lcolladotor@gmail.com" class="email">lcolladotor@gmail.com</a>
      </address>
                  
            <h4 data-toc-skip class="date">27 July 2023</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/LieberInstitute/qsvaRPkgDemo/blob/HEAD/vignettes/qsvaRPkgDemo.Rmd" class="external-link"><code>vignettes/qsvaRPkgDemo.Rmd</code></a></small>
      <div class="hidden name"><code>qsvaRPkgDemo.Rmd</code></div>

    </div>

    
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>Postmortem brain tissue is a valuable resource for studying the human
brain and understanding various neurological and psychiatric disorders.
However, the quality of postmortem brain tissue can be significantly
affected by degradation, which can occur due to various factors such as
the time elapsed since death (postmortem interval), the conditions in
which the body was stored, and the method of tissue preservation.</p>
<p>Therefore, it’s crucial to remove or at least minimize the effect of
degradation in postmortem brain tissue to ensure the accuracy and
reliability of research findings. This can be achieved through various
methods, such as optimizing the conditions for tissue storage and
preservation, using appropriate controls, and using statistical methods
to correct for the effects of degradation.</p>
<p>This method has been described in the initial qSVA framework paper
(Jaffe et al., 2017).</p>
<p>This is a basic example which shows how to obtain the quality
surrogate variables (qSVs) for BrainSeq Phase 2.</p>
</div>
<div class="section level2">
<h2 id="rbioconductor-packages-used">R/Bioconductor packages used<a class="anchor" aria-label="anchor" href="#rbioconductor-packages-used"></a>
</h2>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://bioconductor.org/packages/SummarizedExperiment" class="external-link">SummarizedExperiment</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/stemangiola/tidySummarizedExperiment" class="external-link">tidySummarizedExperiment</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://tidyverse.tidyverse.org" class="external-link">tidyverse</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="http://bioconductor.org/packages/variancePartition" class="external-link">variancePartition</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/jokergoo/ComplexHeatmap" class="external-link">ComplexHeatmap</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/jokergoo/colorRamp2" class="external-link">colorRamp2</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://magrittr.tidyverse.org" class="external-link">magrittr</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://purrr.tidyverse.org/" class="external-link">purrr</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="http://bioinf.wehi.edu.au/limma" class="external-link">limma</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/LieberInstitute/qsvaR" class="external-link">qsvaR</a></span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="time-outline">Time Outline<a class="anchor" aria-label="anchor" href="#time-outline"></a>
</h2>
<table class="table">
<thead><tr class="header">
<th align="center">Activity</th>
<th align="center">Time</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="center">Introduction to qsvaR (slides)</td>
<td align="center">10m</td>
</tr>
<tr class="even">
<td align="center">Data overview</td>
<td align="center">5m</td>
</tr>
<tr class="odd">
<td align="center">Limma and qsvaR</td>
<td align="center">20m</td>
</tr>
</tbody>
</table>
<p>Total: a 35 minute session.</p>
</div>
<div class="section level2">
<h2 id="workshop-goals">Workshop goals<a class="anchor" aria-label="anchor" href="#workshop-goals"></a>
</h2>
<p>During the workshop, we will provide a step-by-step explanation of
the qsvaR and apply it to BrainSeq Phase 2 dataset. We will describe how
using qsvaR can improve the reproducibility of differential expression
analyses across datasets.</p>
</div>
<div class="section level2">
<h2 id="brainseq-phase-2-dataset">BrainSeq Phase 2 Dataset<a class="anchor" aria-label="anchor" href="#brainseq-phase-2-dataset"></a>
</h2>
<p>For illustrative purposes, we’ll use the BrainSeq Phase 2 data at the
transcript level (Collado-Torres et al., 2019). This dataset is RiboZero
RNA-seq data for 900 samples across both the dorsolateral prefrontal
cortex (DLPFC) and the hippocampus (HIPPO) for 551 individuals (286
affected by schizophrenia disorder: SCZD) . Visit <a href="https://eqtl.brainseq.org/phase2/" class="external-link">here</a> for more details.</p>
<div class="section level3">
<h3 id="load-the-data">Load the data<a class="anchor" aria-label="anchor" href="#load-the-data"></a>
</h3>
<p>The BrainSeq Phase 2 data at the transcript level resides in a
<code>RangedSummarizedExperiment</code> (RSE) object called
<code>rse_tx</code>. It has been obtained after running the
<code>SPEAQeasy</code> pipeline. To have a better understanding of how
the pipeline works please attend the workshop that will be presented by
Lieber collegues at the <a href="https://bioc2023.bioconductor.org/abstracts/paper24/" class="external-link">BioC2023</a>.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/options.html" class="external-link">options</a></span><span class="op">(</span>timeout <span class="op">=</span> <span class="fl">400</span><span class="op">)</span></span>
<span><span class="co">## Load the rse_tx object</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/load.html" class="external-link">load</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/connections.html" class="external-link">url</a></span><span class="op">(</span><span class="st">"https://s3.us-east-2.amazonaws.com/libd-brainseq2/rse_tx_unfiltered.Rdata"</span><span class="op">)</span>, verbose <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co">#&gt; Loading objects:</span></span>
<span><span class="co">#&gt;   rse_tx</span></span>
<span></span>
<span><span class="co">## General overview of the object</span></span>
<span><span class="va">rse_tx</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A SummarizedExperiment-tibble abstraction: 178,283,700 × 900</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># </span><span style="color: #555555;">Features=198093 | Samples=900 | Assays=tpm</span></span></span>
<span><span class="co">#&gt;    .feature        .sample   tpm SAMPLE_ID FQCbasicStats perBaseQual perTileQual</span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>           <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;list&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;list&gt;</span>        <span style="color: #949494; font-style: italic;">&lt;list&gt;</span>      <span style="color: #949494; font-style: italic;">&lt;list&gt;</span>     </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> ENST0000045632… R10424   0    <span style="color: #949494;">&lt;chr [1]&gt;</span> <span style="color: #949494;">&lt;chr [1]&gt;</span>     <span style="color: #949494;">&lt;chr [1]&gt;</span>   <span style="color: #949494;">&lt;chr [1]&gt;</span>  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> ENST0000045030… R10424   0    <span style="color: #949494;">&lt;chr [1]&gt;</span> <span style="color: #949494;">&lt;chr [1]&gt;</span>     <span style="color: #949494;">&lt;chr [1]&gt;</span>   <span style="color: #949494;">&lt;chr [1]&gt;</span>  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> ENST0000048814… R10424   2.36 <span style="color: #949494;">&lt;chr [1]&gt;</span> <span style="color: #949494;">&lt;chr [1]&gt;</span>     <span style="color: #949494;">&lt;chr [1]&gt;</span>   <span style="color: #949494;">&lt;chr [1]&gt;</span>  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> ENST0000061921… R10424   0    <span style="color: #949494;">&lt;chr [1]&gt;</span> <span style="color: #949494;">&lt;chr [1]&gt;</span>     <span style="color: #949494;">&lt;chr [1]&gt;</span>   <span style="color: #949494;">&lt;chr [1]&gt;</span>  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> ENST0000047335… R10424   0    <span style="color: #949494;">&lt;chr [1]&gt;</span> <span style="color: #949494;">&lt;chr [1]&gt;</span>     <span style="color: #949494;">&lt;chr [1]&gt;</span>   <span style="color: #949494;">&lt;chr [1]&gt;</span>  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> ENST0000046928… R10424   0    <span style="color: #949494;">&lt;chr [1]&gt;</span> <span style="color: #949494;">&lt;chr [1]&gt;</span>     <span style="color: #949494;">&lt;chr [1]&gt;</span>   <span style="color: #949494;">&lt;chr [1]&gt;</span>  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> ENST0000060709… R10424   0    <span style="color: #949494;">&lt;chr [1]&gt;</span> <span style="color: #949494;">&lt;chr [1]&gt;</span>     <span style="color: #949494;">&lt;chr [1]&gt;</span>   <span style="color: #949494;">&lt;chr [1]&gt;</span>  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> ENST0000041732… R10424   0    <span style="color: #949494;">&lt;chr [1]&gt;</span> <span style="color: #949494;">&lt;chr [1]&gt;</span>     <span style="color: #949494;">&lt;chr [1]&gt;</span>   <span style="color: #949494;">&lt;chr [1]&gt;</span>  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> ENST0000046146… R10424   2.96 <span style="color: #949494;">&lt;chr [1]&gt;</span> <span style="color: #949494;">&lt;chr [1]&gt;</span>     <span style="color: #949494;">&lt;chr [1]&gt;</span>   <span style="color: #949494;">&lt;chr [1]&gt;</span>  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> ENST0000060685… R10424   0    <span style="color: #949494;">&lt;chr [1]&gt;</span> <span style="color: #949494;">&lt;chr [1]&gt;</span>     <span style="color: #949494;">&lt;chr [1]&gt;</span>   <span style="color: #949494;">&lt;chr [1]&gt;</span>  </span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 40 more rows</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 79 more variables: perSeqQual &lt;list&gt;, perBaseContent &lt;list&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   GCcontent &lt;list&gt;, Ncontent &lt;list&gt;, SeqLengthDist &lt;list&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   SeqDuplication &lt;list&gt;, OverrepSeqs &lt;list&gt;, AdapterContent &lt;list&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   KmerContent &lt;list&gt;, percentGC_R1 &lt;list&gt;, phred100_R1 &lt;list&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   phredGT30_R1 &lt;list&gt;, phredGT35_R1 &lt;list&gt;, Adapter88_R1 &lt;list&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   percentGC_R2 &lt;list&gt;, phred100_R2 &lt;list&gt;, phredGT30_R2 &lt;list&gt;, …</span></span></span></code></pre></div>
<p>This is a <code>SummarizedExperiment</code> object but it is
evaluated as a tibble. So it is fully compatible both with
<code>SummarizedExperiment</code> and tidyverse APIs.</p>
</div>
<div class="section level3">
<h3 id="background">Background<a class="anchor" aria-label="anchor" href="#background"></a>
</h3>
<div class="section level4">
<h4 id="summarizedexperiment">SummarizedExperiment<a class="anchor" aria-label="anchor" href="#summarizedexperiment"></a>
</h4>
<p>The <code>SummarizedExperiment</code> package contains two classes:
<code>SummarizedExperiment</code> and
<code>RangedSummarizedExperiment</code>.</p>
<p><code>RangedSummarizedExperiment</code> is a subclass of the
SummarizedExperiment class, meaning it inherits all the features and
functionalities of its parent class.</p>
<div class="alert alert-info">
<p><strong>Note:</strong> The primary difference is that the rows of a
<code>RangedSummarizedExperiment</code> object represent genomic ranges
of interest, rather than a <code>DataFrame</code> of features.</p>
</div>
<p><code>RangedSummarizedExperiment</code> objects use a
<code>GRanges</code> or <code>GRangesList</code> object to represent
genomic ranges. These objects store information about genomic
coordinates, genomic features, and annotations for the ranges of
interest. The <code><a href="https://rdrr.io/pkg/MatrixGenerics/man/rowRanges.html" class="external-link">rowRanges()</a></code> function allows you to access and
manipulate the <code>GenomicRanges</code> information within a
<code>RangedSummarizedExperiment</code> object.</p>
<p>The following graphic displays the class geometry and highlights the
vertical (column) and horizontal (row) relationships.</p>
<div class="figure">
<img src="figs/rse_object.png" style="width:400.0%" alt=""><p class="caption">RangedSummarizedExperiment</p>
</div>
<p><a href="https://bioconductor.org/packages/release/bioc/vignettes/SummarizedExperiment/inst/doc/SummarizedExperiment.html" class="external-link">Source</a></p>
</div>
<div class="section level4">
<h4 id="sample-information">Sample Information<a class="anchor" aria-label="anchor" href="#sample-information"></a>
</h4>
<p>The sample information in <code>colData(rse_tx)</code> contains the
following main variables:</p>
<ul>
<li><p><code>SAMPLE_ID</code> : is the name of the sample.</p></li>
<li><p><code>ERCCsumLogErr</code> : a summary statistic quantifying
overall difference of expected and actual ERCC concentrations for one
sample.</p></li>
<li><p><code>numReads</code> : Total number of reads in fastq
file.</p></li>
<li><p><code>numMapped</code> : Number of reads that were aligned by
HISAT2 to the reference genome.</p></li>
<li><p><code>numUnmapped</code> : Number of reads that did not align to
the reference genome.</p></li>
<li><p><code>overallMapRate</code> : Proportion of total reads mapped
(numMapped / numReads).</p></li>
<li><p><code>concordMapRate</code> : If library is paired-end,
percentage of read pairs that mapped concordantly.</p></li>
<li><p><code>overallMapRate</code>: the decimal fraction of reads which
successfully mapped to the reference genome
(i.e.numMapped/numReads).</p></li>
<li><p><code>totalMapped</code>: the number of reads which successfully
mapped to the canonical sequences in the reference genome (excluding
mitochondrial chromosomes).</p></li>
<li><p><code>mitoMapped</code>: the number of reads which successfully
mapped to the mitochondrial chromosome.</p></li>
<li><p><code>mitoRate</code> : the decimal fraction of reads which
mapped to the mitochondrial chromosome, of those which map at all
(i.e.mitoMapped/ (totalMapped+mitoMapped)).</p></li>
<li><p><code>totalAssignedGene</code> : the decimal fraction of reads
assigned unambiguously to a gene (including mitochondrial genes), with
featureCounts of those in total.</p></li>
<li><p><code>rRNA_rate</code> : the decimal fraction of reads assigned
to a gene whose type is ‘rRNA’, of those assigned to any gene.</p></li>
<li><p><code>RNum</code> : is the identifier for the RNAseq experiment
and is related to the Sample ID.</p></li>
<li><p><code>BrNum</code> : stands for Brain Number. It likely refers to
a unique identifier assigned to each brain (or brain sample) examined in
the study.</p></li>
<li><p><code>Region</code> : refers to a specific region of the brain
from which the sample was taken.</p></li>
<li><p><code>RIN</code> : refers to RNA Integrity Number. It’s a measure
of the quality of the RNA samples used in the study. A higher RIN
indicates higher quality, and thus more reliable, RNA.</p></li>
<li><p><code>Age</code> : refers to the age of the donors from which the
samples were taken.</p></li>
<li><p><code>Sex</code> : refers to the sex of the donors from which the
samples were taken. if the sample comes from a female (F) or a male (M)
donor.</p></li>
<li><p><code>Race</code> : refers to the race of the donors from which
the samples were taken.</p></li>
<li><p><code>Dx</code> : refers to the diagnosis of the donors from
which the samples were taken.</p></li>
</ul>
</div>
<div class="section level4">
<h4 id="data-normalization">Data normalization<a class="anchor" aria-label="anchor" href="#data-normalization"></a>
</h4>
<p>We would like to use the <code>Transcripts Per Million (TPM)</code>
or <code>tpm</code> data. The <code>rse_tx</code> object contains the
following <code>tpm</code> normalized assays:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assays</a></span><span class="op">(</span><span class="va">rse_tx</span><span class="op">)</span></span></code></pre></div>
<p><code>tpm</code>: normalized read counts of the 19,8093 genes across
900 samples.</p>
<div class="alert alert-info">
<p><strong>Note:</strong> <code>Transcripts Per Million (TPM)</code> is
a normalization method for RNA-seq data that facilitates comparisons
both within and between samples. It adjusts for <code>gene length</code>
and <code>sequencing depth</code>, making it a helpful measure for
relative gene expression.</p>
</div>
<p>There are many ways to access the <code>tpm</code> data. Below some
of them:</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Extract the first element from the list of assays in the SummarizedExperiment object</span></span>
<span><span class="co"># This assumes that the first element in the list of assays is the one you are interested in</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assays</a></span><span class="op">(</span><span class="va">rse_tx</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://magrittr.tidyverse.org/reference/aliases.html" class="external-link">extract2</a></span><span class="op">(</span><span class="fl">1</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Another way of extracting the first element from the list of assays in the SummarizedExperiment object</span></span>
<span><span class="co"># Here, 'tpm' is assumed to be the name of the first assay</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assays</a></span><span class="op">(</span><span class="va">rse_tx</span><span class="op">)</span><span class="op">$</span><span class="va">tpm</span></span>
<span></span>
<span><span class="co"># Yet another way of extracting the first element from the list of assays in the SummarizedExperiment object</span></span>
<span><span class="co"># The simplify() function tries to simplify the result to a vector or matrix if possible</span></span>
<span><span class="co"># The first() function extracts the first element of the simplified result</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assays</a></span><span class="op">(</span><span class="va">rse_tx</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://purrr.tidyverse.org/reference/as_vector.html" class="external-link">simplify</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/nth.html" class="external-link">first</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="data-distribution">Data distribution<a class="anchor" aria-label="anchor" href="#data-distribution"></a>
</h4>
<p>Let’s have a look at the distribution of the <code>tpm</code> assay.
Before we have to reformat the data as follows:</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/options.html" class="external-link">options</a></span><span class="op">(</span>timeout <span class="op">=</span> <span class="fl">400</span><span class="op">)</span></span>
<span><span class="co"># Transform 'tpm' data to a long format with 'Transcript', 'Sample', and 'TPM' columns</span></span>
<span><span class="va">tpm_tb</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log2</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assays</a></span><span class="op">(</span><span class="va">rse_tx</span><span class="op">)</span><span class="op">$</span><span class="va">tpm</span> <span class="op">+</span> <span class="fl">0.5</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html" class="external-link">rownames_to_column</a></span><span class="op">(</span>var <span class="op">=</span> <span class="st">"Transcript"</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://tidyr.tidyverse.org/reference/pivot_longer.html" class="external-link">pivot_longer</a></span><span class="op">(</span></span>
<span>        cols <span class="op">=</span> <span class="op">-</span><span class="va">Transcript</span>,</span>
<span>        names_to <span class="op">=</span> <span class="st">"Sample"</span>,</span>
<span>        values_to <span class="op">=</span> <span class="st">"TPM"</span></span>
<span>    <span class="op">)</span></span>
<span></span>
<span><span class="co"># Extract 'RNum', 'Region', and 'Dx' from 'rse_tx', convert to tibble</span></span>
<span><span class="va">colData_tb</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">rse_tx</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">RNum</span>, <span class="va">Region</span>, <span class="va">Dx</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://tibble.tidyverse.org/reference/as_tibble.html" class="external-link">as_tibble</a></span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Join 'tpm_tb' with 'colData_tb' based on 'Sample' and 'RNum'</span></span>
<span><span class="va">tpm_tb</span> <span class="op">&lt;-</span> <span class="va">tpm_tb</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate-joins.html" class="external-link">left_join</a></span><span class="op">(</span><span class="va">colData_tb</span>, by <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Sample"</span> <span class="op">=</span> <span class="st">"RNum"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Calculate mean TPM for each Dx and Region</span></span>
<span><span class="va">mean_tpm</span> <span class="op">&lt;-</span> <span class="va">tpm_tb</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html" class="external-link">group_by</a></span><span class="op">(</span><span class="va">Dx</span>, <span class="va">Region</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/summarise.html" class="external-link">summarise</a></span><span class="op">(</span></span>
<span>        mean_TPM <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/mean.html" class="external-link">mean</a></span><span class="op">(</span><span class="va">TPM</span>, na.rm <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>,</span>
<span>        .groups <span class="op">=</span> <span class="st">"drop"</span> <span class="co"># This drops the grouping</span></span>
<span>    <span class="op">)</span></span></code></pre></div>
<p>Now we can plot the data. Let’s have a look at the boxplots,
histograms and density plots. We’ll display only the boxplots.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Create boxplot for first 100,000 rows, without outliers, split by Dx and Region</span></span>
<span><span class="va">p1</span> <span class="op">&lt;-</span> <span class="va">tpm_tb</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/slice.html" class="external-link">slice</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">100000</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">Dx</span>, y <span class="op">=</span> <span class="va">TPM</span>, fill <span class="op">=</span> <span class="va">Dx</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_boxplot.html" class="external-link">geom_boxplot</a></span><span class="op">(</span>outlier.shape <span class="op">=</span> <span class="cn">NA</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html" class="external-link">facet_wrap</a></span><span class="op">(</span><span class="op">~</span><span class="va">Region</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_minimal</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span>title <span class="op">=</span> <span class="st">"Boxplot of TPM for Each Dx by Region"</span>, x <span class="op">=</span> <span class="st">"Dx"</span>, y <span class="op">=</span> <span class="st">"log2(TPM)+0.5"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Create a histogram of TPM for each Dx</span></span>
<span><span class="va">p2</span> <span class="op">&lt;-</span> <span class="va">tpm_tb</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/slice.html" class="external-link">slice</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">100000</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">TPM</span>, fill <span class="op">=</span> <span class="va">Dx</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_histogram.html" class="external-link">geom_histogram</a></span><span class="op">(</span>alpha <span class="op">=</span> <span class="fl">0.5</span>, bins <span class="op">=</span> <span class="fl">30</span>, position <span class="op">=</span> <span class="st">"identity"</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html" class="external-link">facet_wrap</a></span><span class="op">(</span><span class="op">~</span><span class="va">Dx</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_minimal</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span>title <span class="op">=</span> <span class="st">"Histogram of TPM for Each Dx"</span>, x <span class="op">=</span> <span class="st">"log2(TPM)+0.5"</span>, y <span class="op">=</span> <span class="st">"Count"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Create a density plot of TPM for each Dx</span></span>
<span><span class="va">p3</span> <span class="op">&lt;-</span> <span class="va">tpm_tb</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/slice.html" class="external-link">slice</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">100000</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">TPM</span>, fill <span class="op">=</span> <span class="va">Dx</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_density.html" class="external-link">geom_density</a></span><span class="op">(</span>alpha <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html" class="external-link">facet_wrap</a></span><span class="op">(</span><span class="op">~</span><span class="va">Dx</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_minimal</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span>title <span class="op">=</span> <span class="st">"Density Plot of TPM for Each Dx"</span>, x <span class="op">=</span> <span class="st">"log2(TPM)+0.5"</span>, y <span class="op">=</span> <span class="st">"Density"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">p1</span></span></code></pre></div>
<p>Here the results of the boxplots.</p>
<div class="figure">
<img src="figs/tpmDistributionRegion.png" style="width:400.0%" alt=""><p class="caption">Boxplots</p>
</div>
</div>
<div class="section level4">
<h4 id="correlation-analysis">Correlation Analysis<a class="anchor" aria-label="anchor" href="#correlation-analysis"></a>
</h4>
<p>In essence, our objective is not just to recognize the factors
contributing to expression variation, but to accurately measure their
individual impacts.</p>
<p>Understanding these relationships will allow us to develop a more
accurate and reliable predictive model.</p>
<p>Thus, evaluating the <code>correlation</code> among sample variables
is an imperative step in our analysis.</p>
<p>The presence of <code>highly correlated</code> variables can
significantly <strong>impact</strong> the <code>robustness</code> of our
model, creating <strong>unstable estimates</strong> of variance
fractions.</p>
<style type="text/css">
<style>
p.comment {
background-color: #DBDBDB;
padding: 10px;
border: 1px solid black;
border-radius: 5px;
font-family: sans-serif;
}
</style>
<p class="comment">
</p>
<p>❓ <strong><em>Why assessing correlation is so
important?</em></strong> When two or more variables are highly
correlated, meaning that one can be linearly predicted from the others
with a substantial degree of accuracy, they essentially contain
<code>redundant information</code>. This redundancy can lead to what is
known as <code>multicollinearity</code>, a situation that can greatly
reduce the performance of the model and lead to overfitting.</p>
Therefore, assessing correlation enables us to
<code>prevent multicollinearity</code>, ensuring a more accurate and
reliable analysis.


<p>Let’s first unlist many of the variables below to make it possible to
work with these variables.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">rse_tx</span><span class="op">)</span><span class="op">$</span><span class="va">RIN</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/numeric.html" class="external-link">as.numeric</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/unlist.html" class="external-link">unlist</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">rse_tx</span><span class="op">)</span><span class="op">$</span><span class="va">RIN</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">rse_tx</span><span class="op">)</span><span class="op">$</span><span class="va">mitoMapped</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/unlist.html" class="external-link">unlist</a></span><span class="op">(</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html" class="external-link">map</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">rse_tx</span><span class="op">)</span><span class="op">$</span><span class="va">mitoMapped</span>, <span class="fl">1</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">rse_tx</span><span class="op">)</span><span class="op">$</span><span class="va">totalAssignedGene</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/unlist.html" class="external-link">unlist</a></span><span class="op">(</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html" class="external-link">map</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">rse_tx</span><span class="op">)</span><span class="op">$</span><span class="va">totalAssignedGene</span>, <span class="fl">1</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">rse_tx</span><span class="op">)</span><span class="op">$</span><span class="va">rRNA_rate</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/unlist.html" class="external-link">unlist</a></span><span class="op">(</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html" class="external-link">map</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">rse_tx</span><span class="op">)</span><span class="op">$</span><span class="va">rRNA_rate</span>, <span class="fl">1</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">rse_tx</span><span class="op">)</span><span class="op">$</span><span class="va">numReads</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/unlist.html" class="external-link">unlist</a></span><span class="op">(</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html" class="external-link">map</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">rse_tx</span><span class="op">)</span><span class="op">$</span><span class="va">numReads</span>, <span class="fl">1</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">rse_tx</span><span class="op">)</span><span class="op">$</span><span class="va">numMapped</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/unlist.html" class="external-link">unlist</a></span><span class="op">(</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html" class="external-link">map</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">rse_tx</span><span class="op">)</span><span class="op">$</span><span class="va">numMapped</span>, <span class="fl">1</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">rse_tx</span><span class="op">)</span><span class="op">$</span><span class="va">numUnmapped</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/unlist.html" class="external-link">unlist</a></span><span class="op">(</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html" class="external-link">map</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">rse_tx</span><span class="op">)</span><span class="op">$</span><span class="va">numUnmapped</span>, <span class="fl">1</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">rse_tx</span><span class="op">)</span><span class="op">$</span><span class="va">overallMapRate</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/unlist.html" class="external-link">unlist</a></span><span class="op">(</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html" class="external-link">map</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">rse_tx</span><span class="op">)</span><span class="op">$</span><span class="va">overallMapRate</span>, <span class="fl">1</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">rse_tx</span><span class="op">)</span><span class="op">$</span><span class="va">concordMapRate</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/unlist.html" class="external-link">unlist</a></span><span class="op">(</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html" class="external-link">map</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">rse_tx</span><span class="op">)</span><span class="op">$</span><span class="va">concordMapRate</span>, <span class="fl">1</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">rse_tx</span><span class="op">)</span><span class="op">$</span><span class="va">totalMapped</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/unlist.html" class="external-link">unlist</a></span><span class="op">(</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html" class="external-link">map</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">rse_tx</span><span class="op">)</span><span class="op">$</span><span class="va">totalMapped</span>, <span class="fl">1</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">rse_tx</span><span class="op">)</span><span class="op">$</span><span class="va">mitoRate</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/unlist.html" class="external-link">unlist</a></span><span class="op">(</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html" class="external-link">map</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">rse_tx</span><span class="op">)</span><span class="op">$</span><span class="va">mitoRate</span>, <span class="fl">1</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">rse_tx</span><span class="op">)</span><span class="op">$</span><span class="va">ERCCsumLogErr</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/unlist.html" class="external-link">unlist</a></span><span class="op">(</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html" class="external-link">map</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">rse_tx</span><span class="op">)</span><span class="op">$</span><span class="va">ERCCsumLogErr</span>, <span class="fl">1</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p>We’ll be using the <code><a href="http://DiseaseNeurogenomics.github.io/variancePartition/reference/canCorPairs.html" class="external-link">canCorPairs()</a></code> from the
<code>variancePartition</code> package which performs canonical
correlation analysis (CCA) between pairs of variables.</p>
<p>In the context of the <code>variancePartition</code> package, the
<code><a href="http://DiseaseNeurogenomics.github.io/variancePartition/reference/canCorPairs.html" class="external-link">canCorPairs()</a></code> computes the
<code>canonical correlations</code> between each pair of variables
specified in the formula argument.</p>
<div class="alert alert-info">
<p><strong>Note:</strong> canCorPairs returns correlation values between
0 and 1. Canonical Correlation Analysis (CCA) returns sqrt(mean(rho^2)),
which is the fraction of the maximum possible correlation. When
comparing a two vectors, or a vector and a matrix, this gives the save
value as the absolute correlation. When comparing two sets of
categorical variables, this is equivalent to Cramer’s V statistic.</p>
</div>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">formula</span> <span class="op">&lt;-</span> <span class="op">~</span> <span class="va">Dx</span> <span class="op">+</span> <span class="va">RIN</span> <span class="op">+</span> <span class="va">Age</span> <span class="op">+</span> <span class="va">Sex</span> <span class="op">+</span> <span class="va">Race</span> <span class="op">+</span> <span class="va">mitoMapped</span> <span class="op">+</span> <span class="va">overallMapRate</span> <span class="op">+</span> <span class="va">mitoRate</span> <span class="op">+</span> <span class="va">rRNA_rate</span> <span class="op">+</span> <span class="va">concordMapRate</span> <span class="op">+</span> <span class="va">totalMapped</span> <span class="op">+</span> <span class="va">totalAssignedGene</span> <span class="op">+</span> <span class="va">Region</span></span>
<span><span class="va">correlation_matrix</span> <span class="op">&lt;-</span> <span class="fu"><a href="http://DiseaseNeurogenomics.github.io/variancePartition/reference/canCorPairs.html" class="external-link">canCorPairs</a></span><span class="op">(</span><span class="va">formula</span>, <span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">rse_tx</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Create the heatmap</span></span>
<span><span class="co"># Create a color function from green to purple</span></span>
<span><span class="va">color_func</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/colorRamp2/man/colorRamp2.html" class="external-link">colorRamp2</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="op">-</span><span class="fl">1</span>, <span class="fl">0</span>, <span class="fl">1</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"darkgreen"</span>, <span class="st">"white"</span>, <span class="st">"purple"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span></span>
<span><span class="co"># Create the heatmap object</span></span>
<span><span class="va">heatmap</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/ComplexHeatmap/man/Heatmap.html" class="external-link">Heatmap</a></span><span class="op">(</span><span class="va">correlation_matrix</span>,</span>
<span>    name <span class="op">=</span> <span class="st">"Correlation"</span>, <span class="co"># Title for the color key</span></span>
<span>    show_row_names <span class="op">=</span> <span class="cn">TRUE</span>, <span class="co"># Show row names</span></span>
<span>    show_column_names <span class="op">=</span> <span class="cn">TRUE</span>, <span class="co"># Show column names</span></span>
<span>    cluster_rows <span class="op">=</span> <span class="cn">FALSE</span>, <span class="co"># Do not cluster rows</span></span>
<span>    cluster_columns <span class="op">=</span> <span class="cn">FALSE</span>, <span class="co"># Do not cluster columns</span></span>
<span>    column_title <span class="op">=</span> <span class="st">"Correlation heatmap"</span>, <span class="co"># Add title to the heatmap</span></span>
<span>    cell_fun <span class="op">=</span> <span class="kw">function</span><span class="op">(</span><span class="va">j</span>, <span class="va">i</span>, <span class="va">x</span>, <span class="va">y</span>, <span class="va">width</span>, <span class="va">height</span>, <span class="va">fill</span><span class="op">)</span> <span class="op">{</span></span>
<span>        <span class="fu"><a href="https://rdrr.io/r/grid/grid.text.html" class="external-link">grid.text</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/sprintf.html" class="external-link">sprintf</a></span><span class="op">(</span><span class="st">"%.2f"</span>, <span class="va">correlation_matrix</span><span class="op">[</span><span class="va">i</span>, <span class="va">j</span><span class="op">]</span><span class="op">)</span>, <span class="va">x</span>, <span class="va">y</span>,</span>
<span>            gp <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/grid/gpar.html" class="external-link">gpar</a></span><span class="op">(</span>fontsize <span class="op">=</span> <span class="fl">10</span><span class="op">)</span></span>
<span>        <span class="op">)</span></span>
<span>    <span class="op">}</span>,</span>
<span>    col <span class="op">=</span> <span class="va">color_func</span>, <span class="co"># Use the custom color function</span></span>
<span>    heatmap_legend_param <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span></span>
<span>        title <span class="op">=</span> <span class="st">"Correlation"</span>, <span class="co"># legend title</span></span>
<span>        color_bar <span class="op">=</span> <span class="st">"continuous"</span>, <span class="co"># type of color bar</span></span>
<span>        legend_direction <span class="op">=</span> <span class="st">"horizontal"</span> <span class="co"># direction of legend</span></span>
<span>    <span class="op">)</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Draw the heatmap</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/ComplexHeatmap/man/draw-dispatch.html" class="external-link">draw</a></span><span class="op">(</span><span class="va">heatmap</span><span class="op">)</span></span></code></pre></div>
<p><img src="qsvaRPkgDemo_files/figure-html/PlotHeatmap-1.png" width="700"></p>
<p>The <code>cell_fun</code> function customizes the display of each
cell in the heatmap by adding the correlation value as text inside the
cell with a specific format and font size.</p>
<p>From the heatmap we can see that there’s a high correlation between
<code>mitoMapped</code> and <code>totalAssignedGene</code>. We can’t say
if it’s a positive or negative correlation since it’s an absolute
correlation.</p>
<div class="alert alert-info">
<p><strong>Note:</strong> One of the strategies to address
multicollinearity is removing some of the highly correlated independent
variables. So we have to keep <code>mitoMapped</code> or
<code>totalAssignedGene</code> in our model.</p>
</div>
</div>
</div>
</div>
<div class="section level2">
<h2 id="analysis">Analysis<a class="anchor" aria-label="anchor" href="#analysis"></a>
</h2>
<div class="section level3">
<h3 id="qsvar-quality-surrogate-variable-analysis">qsvaR: Quality Surrogate Variable Analysis<a class="anchor" aria-label="anchor" href="#qsvar-quality-surrogate-variable-analysis"></a>
</h3>
<p>Here’s a general outline of how qSVA can be used to adjust for
degradation in postmortem brain tissue:</p>
<ol style="list-style-type: decimal">
<li><p><strong>Identify Surrogate Variables</strong>: The first step in
qSVA is to identify surrogate variables that capture the
<code>unwanted variation</code> in the data. These variables are often
not of primary interest in the study but can significantly affect the
results.</p></li>
<li><p><strong>Estimate the Effect of Surrogate Variables</strong>: The
next step is to estimate the effect of the surrogate variables on the
data. This is typically done using a statistical model that includes the
surrogate variables as covariates. The coefficients for the surrogate
variables in this model represent their effect on the data.</p></li>
<li><p><strong>Adjust the Data for the Effect of Surrogate Variables in
the DE analysis</strong>: Once the effect of the surrogate variables has
been estimated, this effect can be removed from the data. This is
typically done by subtracting the estimated effect of the surrogate
variables from the data. The resulting adjusted data should be free of
the unwanted variation due to degradation.</p></li>
<li><p><strong>Perform the Primary Analysis on the Adjusted
Data</strong>: Once the data has been adjusted for the effect of the
surrogate variables, the primary analysis can be performed on the
adjusted data. This analysis should yield results that are not
confounded by the unwanted variation due to degradation.</p></li>
</ol>
<div class="section level4">
<h4 id="get-the-degradation-transcripts">Get the degradation transcripts<a class="anchor" aria-label="anchor" href="#get-the-degradation-transcripts"></a>
</h4>
<p>One of the first steps of the <code>qSVA</code> is to get the
degradation transcripts to estimate their effects.</p>
<p>We’ll need to choose one of the three models to get those
transcripts. Here the names <code>cell_component</code>,
<code>top1500</code>, and <code>standard</code> refer to models that
were determined to be effective in removing degradation effects.</p>
</div>
<div class="section level4">
<h4 id="qsvar-models">qsvaR models<a class="anchor" aria-label="anchor" href="#qsvar-models"></a>
</h4>
<div class="section level5">
<h5 id="the-standard-model">The standard model<a class="anchor" aria-label="anchor" href="#the-standard-model"></a>
</h5>
<p>The <code>standard</code> model involves taking the union of the
<code>top 1000</code> transcripts associated with degradation from the
interaction model and the main effect model.</p>
</div>
<div class="section level5">
<h5 id="the-top1500-model">The top1500 model<a class="anchor" aria-label="anchor" href="#the-top1500-model"></a>
</h5>
<p>The <code>top1500</code> model is the same as the “standard” model
except the union of the <code>top 1500</code> genes associated with
degradation is selected.</p>
</div>
<div class="section level5">
<h5 id="the-cell-component-model">The cell component model<a class="anchor" aria-label="anchor" href="#the-cell-component-model"></a>
</h5>
<p>The <code>cell_component</code> model, involved
<code>deconvolution</code> of the degradation matrix to determine the
proportion of cell types within our studied tissue. These proportions
were then added to our <code><a href="https://rdrr.io/r/stats/model.matrix.html" class="external-link">model.matrix()</a></code> and the union of the
top 1000 transcripts in the interaction model, the main effect model,
and the cell proportions model were used to generate this model of qSVs.
<strong>It’s the most effective of our models</strong>.</p>
</div>
</div>
<div class="section level4">
<h4 id="run-qsva-step-by-step">Run qSVA step by step<a class="anchor" aria-label="anchor" href="#run-qsva-step-by-step"></a>
</h4>
<p>After loading the <code>rse_tx</code> object we subset for the
transcripts associated with degradation.</p>
<p>In this example we will choose <code>cell_component</code> when using
the <code><a href="https://rdrr.io/pkg/qsvaR/man/getDegTx.html" class="external-link">getDegTx()</a></code> and <code><a href="https://rdrr.io/pkg/qsvaR/man/select_transcripts.html" class="external-link">select_transcripts()</a></code>
functions. <code><a href="https://rdrr.io/pkg/qsvaR/man/getDegTx.html" class="external-link">getDegTx()</a></code> is used to select a specific set of
transcripts from a given experiment, and provides a warning if the mean
expression level of these transcripts is low.</p>
<p>For a matter of simplification we’ll focus on the dorsolateral
prefrontal cortex (DLPFC) region.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Subset the rse_tx object by DLPFC region</span></span>
<span><span class="va">rse_tx</span> <span class="op">&lt;-</span> <span class="va">rse_tx</span><span class="op">[</span>, <span class="va">rse_tx</span><span class="op">$</span><span class="va">Region</span> <span class="op">==</span> <span class="st">"DLPFC"</span><span class="op">]</span></span>
<span><span class="co">## Next we get the degraded transcripts for qSVA from the "cell_component" model</span></span>
<span><span class="va">DegTx</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/qsvaR/man/getDegTx.html" class="external-link">getDegTx</a></span><span class="op">(</span><span class="va">rse_tx</span>, type <span class="op">=</span> <span class="st">"cell_component"</span><span class="op">)</span></span>
<span><span class="co">## Get the dimension of the data</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">DegTx</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 2976  453</span></span></code></pre></div>
<p>The <code><a href="https://rdrr.io/pkg/qsvaR/man/getPCs.html" class="external-link">getPCs()</a></code> applies PCA using the <code>prcomp</code>
function.</p>
<p><code>prcomp</code> computes principal components by using the
<code>singular value decomposition</code> of the (possibly centered
and/or scaled) data matrix, not by using the covariance matrix. This
makes it <code>more numerically accurate</code> and can be an advantage
when dealing with data that suffers from
<code>multicollinearity</code>.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Now we can compute the Principal Components (PCs) of the degraded transcripts</span></span>
<span><span class="va">pcTx</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/qsvaR/man/getPCs.html" class="external-link">getPCs</a></span><span class="op">(</span><span class="va">DegTx</span>, <span class="st">"tpm"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">pcTx</span><span class="op">$</span><span class="va">x</span><span class="op">)</span><span class="op">[</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">10</span><span class="op">]</span></span>
<span><span class="co">#&gt;               PC1       PC2        PC3       PC4       PC5        PC6       PC7</span></span>
<span><span class="co">#&gt; R10424 -18.609350  5.654972 -10.521455 11.383754 -5.720547 11.1699710 -4.914021</span></span>
<span><span class="co">#&gt; R12195   4.238017 -3.434001  -5.301963  0.755135 -3.632545  5.0044620  2.355794</span></span>
<span><span class="co">#&gt; R12198  15.009249 -6.268986  -6.003985  4.821451 -1.488256  1.3719040 -5.349268</span></span>
<span><span class="co">#&gt; R12199  14.392927 -6.370502  -6.480586  3.787386 -4.982039  4.7914737 -1.066185</span></span>
<span><span class="co">#&gt; R12200 -15.654543  2.845852  -2.273977  3.961545  2.944993  9.3164761  1.634647</span></span>
<span><span class="co">#&gt; R12258   7.109071 -4.316821   7.151549  5.239148 -7.744696 -0.9355408 -1.879552</span></span>
<span><span class="co">#&gt;               PC8         PC9       PC10</span></span>
<span><span class="co">#&gt; R10424 -0.1833597  0.02920567 -1.6005800</span></span>
<span><span class="co">#&gt; R12195 -5.5289264 -1.96673143 -0.1303645</span></span>
<span><span class="co">#&gt; R12198  2.8738392 -0.64170546 -1.3889163</span></span>
<span><span class="co">#&gt; R12199 -2.7378677  2.26850143  1.8711155</span></span>
<span><span class="co">#&gt; R12200 -9.1117425 -2.32982944  3.6057435</span></span>
<span><span class="co">#&gt; R12258  8.1395165 -2.19594082 -0.7708335</span></span></code></pre></div>
<p>The <code>prcomp</code> function in R performs a Principal Component
Analysis (PCA) and returns a list with several components. Here’s what
the x component represents:</p>
<p>x: this component will contain the original data projected onto the
principal components (also known as scores). This will be a matrix where
each row corresponds to an observation in the original data and each
column corresponds to a principal component.</p>
<p>After the principal component analysis we creates a model matrix for
regression analysis using a set of predictors.</p>
<p>We use a model matrix that accounts for relevant variables (Charity
et al., 2020). We include variables such as Age, Sex and Race in this
model. Our dataset, <code>DegTx</code>, a
<code>RangedSummarizedExperiment</code> object, is used as the
<code>rse_tx</code> input.</p>
<div class="alert alert-info">
<p><strong>Note:</strong> The <code><a href="https://rdrr.io/r/stats/model.matrix.html" class="external-link">model.matrix()</a></code> takes a
formula and a data frame (or similar object) and returns the design
matrix.</p>
<p><code><a href="https://rdrr.io/r/stats/model.matrix.html" class="external-link">model.matrix()</a></code> will automatically create dummy variables
for each level of the factor. Dummy variables are binary (0/1) variables
that indicate the presence of a categorical level for each
observation.</p>
</div>
<p>Let’s understand this step by step:</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Creates a model matrix for regression analysis using the predictors Dx, Age, Sex and Race from the column data of the rse_tx object.</span></span>
<span><span class="va">rse_tx</span><span class="op">$</span><span class="va">Dx</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">as.factor</a></span><span class="op">(</span><span class="va">rse_tx</span><span class="op">$</span><span class="va">Dx</span><span class="op">)</span></span>
<span><span class="va">mod</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/model.matrix.html" class="external-link">model.matrix</a></span><span class="op">(</span><span class="op">~</span> <span class="va">Dx</span> <span class="op">+</span> <span class="va">Age</span> <span class="op">+</span> <span class="va">Sex</span> <span class="op">+</span> <span class="va">Race</span>,</span>
<span>    data <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">rse_tx</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># check the names of the columns in the mod matrix</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">mod</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "(Intercept)" "DxSchizo"    "Age"         "SexM"        "RaceAS"     </span></span>
<span><span class="co">#&gt; [6] "RaceCAUC"    "RaceHISP"</span></span></code></pre></div>
<p>The <code><a href="https://rdrr.io/r/stats/model.matrix.html" class="external-link">model.matrix()</a></code> takes the variables in the formula
format:</p>
<p><code>model.matrix([target] ~ [predictor / features], data = [data source])</code></p>
<p>This part is called a formula:
<code>~ Dx + Age + Sex + Race</code></p>
<p>The tilde (~) can be read as <code>is modeled as</code>. The
variables on the right side of the tilde are the predictor variables (or
features).</p>
<p>In this case, Dx, Age, Sex and Race are your
<code>predictors</code>.</p>
<p>The <code><a href="https://rdrr.io/pkg/qsvaR/man/k_qsvs.html" class="external-link">k_qsvs()</a></code> function is employed to determine the
required number of Principal Components (PCs) needed to capture the
underlying variation in our data.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Get the number of qsvs</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">20230703</span><span class="op">)</span></span>
<span><span class="va">k</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/qsvaR/man/k_qsvs.html" class="external-link">k_qsvs</a></span><span class="op">(</span><span class="va">DegTx</span>, <span class="va">mod</span>, <span class="st">"tpm"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">k</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 25</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="run-qsva-using-the-qsva-wrapper-function">Run qSVA using the qSVA wrapper function<a class="anchor" aria-label="anchor" href="#run-qsva-using-the-qsva-wrapper-function"></a>
</h4>
<p>You can simplify the process by utilizing our integrated function,
<code>qSVA</code>. This wrapper function conveniently combines all
previously mentioned functions, streamlining the analysis into a single
step.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Example use of the wrapper function qSVA()</span></span>
<span><span class="co">## set.seed() (in this case, 20230703) is an integer that R uses to initialize its random number generator</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">20230703</span><span class="op">)</span></span>
<span><span class="va">qsva_pcs_cc</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/qsvaR/man/qSVA.html" class="external-link">qSVA</a></span><span class="op">(</span><span class="va">rse_tx</span>, type <span class="op">=</span> <span class="st">"cell_component"</span>, mod <span class="op">=</span> <span class="va">mod</span>, assayname <span class="op">=</span> <span class="st">"tpm"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">qsva_pcs_cc</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 453  25</span></span></code></pre></div>
</div>
</div>
<div class="section level3">
<h3 id="differential-expression-analysis-using-limma-between-the-schizophrenia-schizo-and-neurotypical-control-groups-control">Differential Expression Analysis using limma between the
schizophrenia (Schizo) and neurotypical control groups (Control)<a class="anchor" aria-label="anchor" href="#differential-expression-analysis-using-limma-between-the-schizophrenia-schizo-and-neurotypical-control-groups-control"></a>
</h3>
<p>Before jumping into the differential expression analysis we have to
add the qSVs to our statistical model by using the
<code><a href="https://rdrr.io/r/base/cbind.html" class="external-link">cbind()</a></code>.</p>
<p><code>mod_qSVA</code> includes both our original model and the qSVs.
The qSVs adjust for the degradation effect.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Add the qSVs to our statistical model</span></span>
<span><span class="va">mod_qSVA</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/cbind.html" class="external-link">cbind</a></span><span class="op">(</span></span>
<span>    <span class="va">mod</span>,</span>
<span>    <span class="va">qsva_pcs_cc</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><code>limma</code> fits a so-called linear model. <a href="https://bioconductor.org/packages/release/bioc/vignettes/limma/inst/doc/usersguide.pdf" class="external-link">Here</a>
the limma User’s Guide from the limma bioconductor webpage.</p>
<p>In the context of limma, each gene’s expression level is modeled as a
linear combination of the experimental factors, with some added noise or
error term.</p>
<p>The <code><a href="https://rdrr.io/pkg/limma/man/lmFit.html" class="external-link">lmFit()</a></code> function fits a separate linear model to
each gene (or transcript, in this case), and the result (fitTx) is an
object that contains the fitted model coefficients and other statistics
for each gene.</p>
<p>After fitting the model, <code><a href="https://rdrr.io/pkg/limma/man/ebayes.html" class="external-link">eBayes()</a></code> is used to compute
<code>empirical Bayes statistics</code> for differential expression. The
function applies a shrinkage technique to the standard errors,
<code>improving the stability of the estimates</code>, particularly when
the <code>number of samples is small</code>.</p>
<p>The object eBTx that results from the <code><a href="https://rdrr.io/pkg/limma/man/ebayes.html" class="external-link">eBayes()</a></code> call is a
fitted model object that contains several components, including
<code>coefficients</code>, <code>standard errors</code>,
<code>t-statistics</code>, <code>p-values</code>, and more.</p>
<div class="alert alert-info">
<p><strong>Note:</strong> Neither <code>TPM</code> nor <code>CPM</code>
should be fed into edgeR or <code>voom</code>, both expects
<code>raw counts</code>, as <code>library sizes</code> are used in the
<code>normalization process</code>.</p>
</div>
<p>The steps are the following:</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Extract the transcript expression values and put them in the</span></span>
<span><span class="co">## log2(TPM + 0.5) scale</span></span>
<span><span class="va">txExprs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log2</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assays</a></span><span class="op">(</span><span class="va">rse_tx</span><span class="op">)</span><span class="op">$</span><span class="va">tpm</span> <span class="op">+</span> <span class="fl">0.5</span><span class="op">)</span></span>
<span><span class="co">## Run the standard linear model for differential expression</span></span>
<span><span class="va">fitTx</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/limma/man/lmFit.html" class="external-link">lmFit</a></span><span class="op">(</span><span class="va">txExprs</span>, <span class="va">mod_qSVA</span><span class="op">)</span></span>
<span><span class="va">eBTx</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/limma/man/ebayes.html" class="external-link">eBayes</a></span><span class="op">(</span><span class="va">fitTx</span><span class="op">)</span></span></code></pre></div>
<p>In this part above of the code, we are running a linear model for
differential expression analysis using the <code><a href="https://rdrr.io/pkg/limma/man/lmFit.html" class="external-link">lmFit()</a></code> from the
<code>limma</code> package in R.</p>
<p>The function takes two main arguments: the matrix of expression
values (txExprs) and the design matrix (mod_qSVA).</p>
<div class="alert alert-info">
<p><strong>Note:</strong> Once we have our <code>mod_qSVA</code> we can
use for the <code>txExprs</code> an <code>rse_gene</code> instead of an
<code>rse_tx</code>.</p>
</div>
<p>The design matrix specifies the linear model that you’re fitting to
the data. In this case, mod_qSVA includes the variables of interest and
the qSVs.</p>
<div class="alert alert-info">
<p><strong>Note:</strong> The <code>colnames(eBTx)</code> function will
return the names of the columns of the coefficients matrix in the eBTx
object. Each column of this matrix corresponds to a coefficient in the
linear model, which in turn corresponds to a comparison of interest in
your experiment.</p>
</div>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Get the comparison names from the fitted model 'eBTx'</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">eBTx</span><span class="op">)</span></span>
<span><span class="co">#&gt;  [1] "(Intercept)" "DxSchizo"    "Age"         "SexM"        "RaceAS"     </span></span>
<span><span class="co">#&gt;  [6] "RaceCAUC"    "RaceHISP"    "qSV1"        "qSV2"        "qSV3"       </span></span>
<span><span class="co">#&gt; [11] "qSV4"        "qSV5"        "qSV6"        "qSV7"        "qSV8"       </span></span>
<span><span class="co">#&gt; [16] "qSV9"        "qSV10"       "qSV11"       "qSV12"       "qSV13"      </span></span>
<span><span class="co">#&gt; [21] "qSV14"       "qSV15"       "qSV16"       "qSV17"       "qSV18"      </span></span>
<span><span class="co">#&gt; [26] "qSV19"       "qSV20"       "qSV21"       "qSV22"       "qSV23"      </span></span>
<span><span class="co">#&gt; [31] "qSV24"       "qSV25"</span></span></code></pre></div>
<p>Let’s extract the differential expression results.</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Extract the differential expression results</span></span>
<span><span class="va">sigTx</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/limma/man/toptable.html" class="external-link">topTable</a></span><span class="op">(</span><span class="va">eBTx</span>,</span>
<span>    coef <span class="op">=</span> <span class="st">"DxSchizo"</span>, <span class="co"># test for differential expression with respect to the "DxSchizo" coefficient.</span></span>
<span>    p.value <span class="op">=</span> <span class="fl">1</span>, <span class="co"># all genes will be returned, regardless of their p-values</span></span>
<span>    number <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">rse_tx</span><span class="op">)</span> <span class="co"># all genes to be returned</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><code>coef = "DxSchizo"</code> specifies the model coefficient of
interest. The <code><a href="https://rdrr.io/pkg/limma/man/toptable.html" class="external-link">topTable()</a></code> will return results for this
specific term.</p>
<p>Let’s explore the top results.</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Explore the top results</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">sigTx</span><span class="op">)</span></span>
<span><span class="co">#&gt;                         logFC    AveExpr         t      P.Value adj.P.Val</span></span>
<span><span class="co">#&gt; ENST00000509833.1 -0.19793326  1.6039735 -4.984710 9.075361e-07 0.1797765</span></span>
<span><span class="co">#&gt; ENST00000206423.7 -0.05451499  1.1566859 -4.765995 2.590207e-06 0.2005694</span></span>
<span><span class="co">#&gt; ENST00000596272.5  0.14845134 -0.6480373  4.732048 3.037503e-06 0.2005694</span></span>
<span><span class="co">#&gt; ENST00000518335.2  0.09767838 -0.5550917  4.631073 4.851566e-06 0.2221915</span></span>
<span><span class="co">#&gt; ENST00000553503.5 -0.30153792  1.0366885 -4.582705 6.053531e-06 0.2221915</span></span>
<span><span class="co">#&gt; ENST00000416708.1  0.07597952 -0.9207563  4.559409 6.729914e-06 0.2221915</span></span>
<span><span class="co">#&gt;                          B</span></span>
<span><span class="co">#&gt; ENST00000509833.1 5.237794</span></span>
<span><span class="co">#&gt; ENST00000206423.7 4.243017</span></span>
<span><span class="co">#&gt; ENST00000596272.5 4.092195</span></span>
<span><span class="co">#&gt; ENST00000518335.2 3.649307</span></span>
<span><span class="co">#&gt; ENST00000553503.5 3.440212</span></span>
<span><span class="co">#&gt; ENST00000416708.1 3.340211</span></span></code></pre></div>
<p>Here’s a breakdown of what each column in this output represents:</p>
<ul>
<li>
<strong>logFC (Log Fold Change)</strong>: This represents the log2
fold-change for the differential expression between two conditions.</li>
</ul>
<div class="alert alert-info">
<p><strong>Note:</strong> A positive value indicates higher expression
in the second condition, while a negative value indicates higher
expression in the first condition. For example, a logFC of -0.07856532
for ENST00000553142.5 means this transcript has lower expression in the
second condition compared to the first.</p>
</div>
<ul>
<li><p><strong>AveExpr (Average Expression)</strong>: This is the
average log2 expression level of the gene across all samples. It
provides a general idea of how abundantly the gene is
expressed.</p></li>
<li><p><strong>t (t-statistic)</strong>: This is the t-statistic from
the differential expression test. It is a measure of how different the
expression levels of the gene are between the two conditions, in units
of standard error.</p></li>
</ul>
<div class="alert alert-info">
<p><strong>Note:</strong> A higher absolute value of the t-statistic
means the gene’s expression is more significantly different between the
conditions.</p>
</div>
<ul>
<li>
<strong>P.Value (p-value)</strong>: This is the p-value from the
differential expression test. It quantifies the statistical evidence
against the null hypothesis that the gene’s expression levels are the
same in the two conditions.</li>
</ul>
<div class="alert alert-info">
<p><strong>Note:</strong> A lower p-value suggests stronger evidence
against the null hypothesis.</p>
</div>
<ul>
<li>
<strong>adj.P.Val (Adjusted p-value)</strong>: This is the p-value
adjusted for multiple testing using the method of Benjamini and
Hochberg, also known as the false discovery rate (FDR). It helps control
for false positives when testing multiple hypotheses (genes) at
once.</li>
</ul>
<div class="alert alert-info">
<p><strong>Note:</strong> Typically, genes with an adjusted p-value (or
FDR) less than 0.05 or 0.01 are considered statistically significantly
differentially expressed.</p>
</div>
<ul>
<li>
<strong>B (B-statistic or log-odds of differential
expression)</strong>: This is the B-statistic from the empirical Bayes
analysis, which can be interpreted as the log-odds that the gene is
differentially expressed.</li>
</ul>
<div class="alert alert-info">
<p><strong>Note:</strong> A higher B-statistic means the gene is more
likely to be differentially expressed.</p>
</div>
</div>
<div class="section level3">
<h3 id="differential-expression-quality-or-dequal-plots">Differential expression quality or DEqual plots<a class="anchor" aria-label="anchor" href="#differential-expression-quality-or-dequal-plots"></a>
</h3>
<p>This R function, <code>DEqual</code>, is designed to compare
<code>degradation t-statistics</code> with the <code>t-statistics</code>
from a differential expression (DE) analysis in the human brain
tissue.</p>
<p>It takes as input a data frame (DE) containing differential
expression results, such as those outputted by the limma package’s
topTable function.</p>
<p>The function identifies the transcripts that are common between the
<code>DE dataframe</code> and <code>degradation_tstats</code> from the
qsvaR package and creates a new dataframe, common_data, which contains
the <code>degradation t-statistics</code> and
<code>DE t-statistics</code> for the common transcripts.</p>
<p>In this plot:</p>
<ul>
<li><p><strong>Dark purple to blue bins</strong>: These colors represent
bins with a low number of data points. If a bin is dark purple or blue,
it means there are relatively few transcripts with those particular DE
and degradation t-statistics.</p></li>
<li><p><strong>Green to yellow bins</strong>: These colors represent
bins with a moderate number of data points. If a bin is green or yellow,
it means there are a moderate number of transcripts with those
particular DE and degradation t-statistics.</p></li>
<li><p><strong>Bright yellow bins</strong>: These colors represent bins
with a high number of data points. If a bin is bright yellow, it means
there are many transcripts with those particular DE and degradation
t-statistics.</p></li>
</ul>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Generate a DEqual() plot using the model results without qSVs</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">20230703</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/qsvaR/man/DEqual.html" class="external-link">DEqual</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/limma/man/toptable.html" class="external-link">topTable</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/limma/man/ebayes.html" class="external-link">eBayes</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/limma/man/lmFit.html" class="external-link">lmFit</a></span><span class="op">(</span><span class="va">txExprs</span>, <span class="va">mod</span><span class="op">)</span><span class="op">)</span>, coef <span class="op">=</span> <span class="st">"DxSchizo"</span>, p.value <span class="op">=</span> <span class="fl">1</span>, number <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">rse_tx</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"DEqual without qSVA"</span><span class="op">)</span></span></code></pre></div>
<p><img src="qsvaRPkgDemo_files/figure-html/DEqualwithout-1.png" width="700"></p>
<p>The first plot reveals a correlation of <strong>0.48</strong> with
the degradation experiment, even after adjusting for several common
variables. Such a high correlation underscores the fact that
degradation’s signal remains significantly intertwined with our data.
This can potentially introduce <code>confounding factors</code> in our
differential expression analysis between the schizophrenia (SCZD) and
neurotypical control groups.</p>
<div class="alert alert-info">
<p><strong>Note:</strong> In essence, the high correlation in the first
plot could be masking the true differential gene expression between the
case-control groups by mingling the degradation signal into the
expression data. This emphasizes the need for adequate normalization,
such as qSVA, to reduce such confounding factors and ensure a more
accurate representation of the biological differences in gene
expression.</p>
</div>
<p>For comparison, here is the DEqual() plot for the model with
qSVs.</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Generate a DEqual() plot using the model results with qSVs</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">20230703</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/qsvaR/man/DEqual.html" class="external-link">DEqual</a></span><span class="op">(</span><span class="va">sigTx</span><span class="op">)</span> <span class="op">+</span></span>
<span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"DEqual with qSVA"</span><span class="op">)</span></span></code></pre></div>
<p><img src="qsvaRPkgDemo_files/figure-html/DEqualwith-1.png" width="700"></p>
<p><br></p>
<p>Conversely, in the second plot, the correlation stands at
<strong>-0.0019</strong>, suggesting a <code>minimal relationship</code>
between degradation and the differential expression results. This
indicates that the <strong>degradation effects have been effectively
eliminated from the data</strong>, leading to a cleaner interpretation
of the gene expression differences.</p>
</div>
</div>
<div class="section level2">
<h2 id="conclusion">Conclusion<a class="anchor" aria-label="anchor" href="#conclusion"></a>
</h2>
<ul>
<li><p>During this workshop, we gained insights and practical knowledge
on how to adjust for degradation in postmortem human brain tissue using
<strong>qsvaR</strong>.</p></li>
<li><p>The results from the differential expression analysis with and
without qSVA normalization shown in the two DEqual plots reveals the
effectiveness of degradation effect removal.</p></li>
<li><p>The use of the qSVA effectively removes degradation
effects.</p></li>
</ul>
</div>
<div class="section level2">
<h2 id="reproducibility">Reproducibility<a class="anchor" aria-label="anchor" href="#reproducibility"></a>
</h2>
<p>The <em><a href="https://bioconductor.org/packages/3.15/qsvaRPkgDemo" class="external-link">qsvaRPkgDemo</a></em>
package (HediaTnani, 2023) was made possible thanks to:</p>
<ul>
<li>R (R Core Team, 2022)</li>
<li>
<em><a href="https://bioconductor.org/packages/3.15/BiocStyle" class="external-link">BiocStyle</a></em>
(Oleś, 2022)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=knitr" class="external-link">knitr</a></em> (Xie,
2023)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=RefManageR" class="external-link">RefManageR</a></em>
(McLean, 2017)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=rmarkdown" class="external-link">rmarkdown</a></em>
(Allaire, Xie, Dervieux, McPherson, Luraschi, Ushey, Atkins, Wickham,
Cheng, Chang, and Iannone, 2023)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=sessioninfo" class="external-link">sessioninfo</a></em>
(Wickham, Chang, Flight, Müller, and Hester, 2021)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=testthat" class="external-link">testthat</a></em>
(Wickham, 2011)</li>
<li>
<em><a href="https://bioconductor.org/packages/3.15/SummarizedExperiment" class="external-link">SummarizedExperiment</a></em>
(Morgan, Obenchain, Hester, and Pagès, 2022)</li>
<li>
<em><a href="https://bioconductor.org/packages/3.15/tidySummarizedExperiment" class="external-link">tidySummarizedExperiment</a></em>
(Mangiola, 2022)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=tidyverse" class="external-link">tidyverse</a></em>
(Wickham, Averick, Bryan, Chang, McGowan, François, Grolemund, Hayes,
Henry, Hester, Kuhn, Pedersen, Miller, Bache, Müller, Ooms, Robinson,
Seidel, Spinu, Takahashi, Vaughan, Wilke, Woo, and Yutani, 2019)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=variancePartition" class="external-link">variancePartition</a></em>
(Hoffman and Schadt, 2016)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=ComplexHeatmap" class="external-link">ComplexHeatmap</a></em>
(Gu, Eils, and Schlesner, 2016)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=colorRamp2" class="external-link">colorRamp2</a></em>
(Gu, 2022)</li>
<li>
<em><a href="https://bioconductor.org/packages/3.15/limma" class="external-link">limma</a></em>
(Ritchie, Phipson, Wu, Hu, Law, Shi, and Smyth, 2015)</li>
<li><em><a href="https://CRAN.R-project.org/package=ggplot2" class="external-link">ggplot2</a></em></li>
<li>
<em><a href="https://CRAN.R-project.org/package=dplyr" class="external-link">dplyr</a></em> (Wickham,
François, Henry, Müller, and Vaughan, 2023)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=purrr" class="external-link">purrr</a></em> (Wickham
and Henry, 2023)</li>
<li><em><a href="https://github.com/jaffelab" class="external-link">jaffelab</a></em></li>
<li>
<em><a href="https://bioconductor.org/packages/3.15/qsvaR" class="external-link">qsvaR</a></em>
(Stolz and Collado-Torres, 2023)</li>
</ul>
<p>This package was developed using <em><a href="https://bioconductor.org/packages/3.15/biocthis" class="external-link">biocthis</a></em>.</p>
<p>Code for creating the vignette</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Create the vignette</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/rstudio/rmarkdown" class="external-link">"rmarkdown"</a></span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/system.time.html" class="external-link">system.time</a></span><span class="op">(</span><span class="fu"><a href="https://pkgs.rstudio.com/rmarkdown/reference/render.html" class="external-link">render</a></span><span class="op">(</span><span class="st">"qsvaRPkgDemo.Rmd"</span>, <span class="st">"BiocStyle::html_document"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Extract the R code</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://yihui.org/knitr/" class="external-link">"knitr"</a></span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/knit.html" class="external-link">knit</a></span><span class="op">(</span><span class="st">"qsvaRPkgDemo.Rmd"</span>, tangle <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<p>Date the vignette was generated.</p>
<pre><code><span><span class="co">#&gt; [1] "2023-07-27 16:45:38 UTC"</span></span></code></pre>
<p>Wallclock time spent generating the vignette.</p>
<pre><code><span><span class="co">#&gt; Time difference of 1.235 mins</span></span></code></pre>
<p><code>R</code> session information.</p>
<pre><code><span><span class="co">#&gt; <span style="color: #00BBBB; font-weight: bold;">─ Session info ───────────────────────────────────────────────────────────────────────────────────────────────────────</span></span></span>
<span><span class="co">#&gt;  <span style="color: #555555; font-style: italic;">setting </span> <span style="color: #555555; font-style: italic;">value</span></span></span>
<span><span class="co">#&gt;  version  R version 4.2.1 (2022-06-23)</span></span>
<span><span class="co">#&gt;  os       Ubuntu 20.04.4 LTS</span></span>
<span><span class="co">#&gt;  system   x86_64, linux-gnu</span></span>
<span><span class="co">#&gt;  ui       X11</span></span>
<span><span class="co">#&gt;  language en</span></span>
<span><span class="co">#&gt;  collate  en_US.UTF-8</span></span>
<span><span class="co">#&gt;  ctype    en_US.UTF-8</span></span>
<span><span class="co">#&gt;  tz       UTC</span></span>
<span><span class="co">#&gt;  date     2023-07-27</span></span>
<span><span class="co">#&gt;  pandoc   2.18 @ /usr/local/bin/ (via rmarkdown)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB; font-weight: bold;">─ Packages ───────────────────────────────────────────────────────────────────────────────────────────────────────────</span></span></span>
<span><span class="co">#&gt;  <span style="color: #555555; font-style: italic;">package                 </span> <span style="color: #555555; font-style: italic;">*</span> <span style="color: #555555; font-style: italic;">version  </span> <span style="color: #555555; font-style: italic;">date (UTC)</span> <span style="color: #555555; font-style: italic;">lib</span> <span style="color: #555555; font-style: italic;">source</span></span></span>
<span><span class="co">#&gt;  annotate                   1.74.0    <span style="color: #555555;">2022-04-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  AnnotationDbi              1.58.0    <span style="color: #555555;">2022-04-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  aod                        1.3.2     <span style="color: #555555;">2022-04-02</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  backports                  1.4.1     <span style="color: #555555;">2021-12-13</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  bibtex                     0.5.1     <span style="color: #555555;">2023-01-26</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  Biobase                  * 2.56.0    <span style="color: #555555;">2022-04-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  BiocGenerics             * 0.42.0    <span style="color: #555555;">2022-04-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  BiocManager                1.30.21.1 <span style="color: #555555;">2023-07-18</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  BiocParallel             * 1.30.4    <span style="color: #555555;">2022-10-11</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  BiocStyle                * 2.24.0    <span style="color: #555555;">2022-04-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  Biostrings                 2.64.1    <span style="color: #555555;">2022-08-18</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  bit                        4.0.5     <span style="color: #555555;">2022-11-15</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  bit64                      4.0.5     <span style="color: #555555;">2020-08-30</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  bitops                     1.0-7     <span style="color: #555555;">2021-04-24</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  blob                       1.2.4     <span style="color: #555555;">2023-03-17</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  bookdown                   0.34      <span style="color: #555555;">2023-05-09</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  boot                       1.3-28.1  <span style="color: #555555;">2022-11-22</span> <span style="color: #555555;">[3]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  broom                      1.0.5     <span style="color: #555555;">2023-06-09</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  bslib                      0.5.0     <span style="color: #555555;">2023-06-09</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  cachem                     1.0.8     <span style="color: #555555;">2023-05-01</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  caTools                    1.18.2    <span style="color: #555555;">2021-03-28</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  circlize                   0.4.15    <span style="color: #555555;">2022-05-10</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  cli                        3.6.1     <span style="color: #555555;">2023-03-23</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  clue                       0.3-64    <span style="color: #555555;">2023-01-31</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  cluster                    2.1.4     <span style="color: #555555;">2022-08-22</span> <span style="color: #555555;">[3]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  codetools                  0.2-19    <span style="color: #555555;">2023-02-01</span> <span style="color: #555555;">[3]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  colorRamp2               * 0.1.0     <span style="color: #555555;">2022-12-21</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  colorspace                 2.1-0     <span style="color: #555555;">2023-01-23</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  ComplexHeatmap           * 2.12.1    <span style="color: #555555;">2022-08-09</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  crayon                     1.5.2     <span style="color: #555555;">2022-09-29</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  data.table                 1.14.8    <span style="color: #555555;">2023-02-17</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  DBI                        1.1.3     <span style="color: #555555;">2022-06-18</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  DelayedArray               0.22.0    <span style="color: #555555;">2022-04-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  desc                       1.4.2     <span style="color: #555555;">2022-09-08</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  digest                     0.6.33    <span style="color: #555555;">2023-07-07</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  doParallel                 1.0.17    <span style="color: #555555;">2022-02-07</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  dplyr                    * 1.1.2     <span style="color: #555555;">2023-04-20</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  edgeR                      3.38.4    <span style="color: #555555;">2022-08-07</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  ellipsis                   0.3.2     <span style="color: #555555;">2021-04-29</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  evaluate                   0.21      <span style="color: #555555;">2023-05-05</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  fansi                      1.0.4     <span style="color: #555555;">2023-01-22</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  farver                     2.1.1     <span style="color: #555555;">2022-07-06</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  fastmap                    1.1.1     <span style="color: #555555;">2023-02-24</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  forcats                  * 1.0.0     <span style="color: #555555;">2023-01-29</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  foreach                    1.5.2     <span style="color: #555555;">2022-02-02</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  fs                         1.6.3     <span style="color: #555555;">2023-07-20</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  genefilter                 1.78.0    <span style="color: #555555;">2022-04-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  generics                   0.1.3     <span style="color: #555555;">2022-07-05</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  GenomeInfoDb             * 1.32.4    <span style="color: #555555;">2022-09-06</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  GenomeInfoDbData           1.2.8     <span style="color: #555555;">2022-05-02</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  GenomicRanges            * 1.48.0    <span style="color: #555555;">2022-04-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  GetoptLong                 1.0.5     <span style="color: #555555;">2020-12-15</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  ggplot2                  * 3.4.2     <span style="color: #555555;">2023-04-03</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  GlobalOptions              0.1.2     <span style="color: #555555;">2020-06-10</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  glue                       1.6.2     <span style="color: #555555;">2022-02-24</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  gplots                     3.1.3     <span style="color: #555555;">2022-04-25</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  gtable                     0.3.3     <span style="color: #555555;">2023-03-21</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  gtools                     3.9.4     <span style="color: #555555;">2022-11-27</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  highr                      0.10      <span style="color: #555555;">2022-12-22</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  hms                        1.1.3     <span style="color: #555555;">2023-03-21</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  htmltools                  0.5.5     <span style="color: #555555;">2023-03-23</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  htmlwidgets                1.6.2     <span style="color: #555555;">2023-03-17</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  httr                       1.4.6     <span style="color: #555555;">2023-05-08</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  IRanges                  * 2.30.1    <span style="color: #555555;">2022-08-18</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  iterators                  1.0.14    <span style="color: #555555;">2022-02-05</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  jquerylib                  0.1.4     <span style="color: #555555;">2021-04-26</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  jsonlite                   1.8.7     <span style="color: #555555;">2023-06-29</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  KEGGREST                   1.36.3    <span style="color: #555555;">2022-07-12</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  KernSmooth                 2.23-22   <span style="color: #555555;">2023-07-10</span> <span style="color: #555555;">[3]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  knitr                      1.43      <span style="color: #555555;">2023-05-25</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  labeling                   0.4.2     <span style="color: #555555;">2020-10-20</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  lattice                    0.21-8    <span style="color: #555555;">2023-04-05</span> <span style="color: #555555;">[3]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  lazyeval                   0.2.2     <span style="color: #555555;">2019-03-15</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  lifecycle                  1.0.3     <span style="color: #555555;">2022-10-07</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  limma                    * 3.52.4    <span style="color: #555555;">2022-09-27</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  lme4                       1.1-34    <span style="color: #555555;">2023-07-04</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  locfit                     1.5-9.8   <span style="color: #555555;">2023-06-11</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  lubridate                * 1.9.2     <span style="color: #555555;">2023-02-10</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  magrittr                 * 2.0.3     <span style="color: #555555;">2022-03-30</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  MASS                       7.3-60    <span style="color: #555555;">2023-05-04</span> <span style="color: #555555;">[3]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  Matrix                     1.6-0     <span style="color: #555555;">2023-07-08</span> <span style="color: #555555;">[3]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  MatrixGenerics           * 1.8.1     <span style="color: #555555;">2022-06-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  matrixStats              * 1.0.0     <span style="color: #555555;">2023-06-02</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  memoise                    2.0.1     <span style="color: #555555;">2021-11-26</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  mgcv                       1.9-0     <span style="color: #555555;">2023-07-11</span> <span style="color: #555555;">[3]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  minqa                      1.2.5     <span style="color: #555555;">2022-10-19</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  munsell                    0.5.0     <span style="color: #555555;">2018-06-12</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  nlme                       3.1-162   <span style="color: #555555;">2023-01-31</span> <span style="color: #555555;">[3]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  nloptr                     2.0.3     <span style="color: #555555;">2022-05-26</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  pbkrtest                   0.5.2     <span style="color: #555555;">2023-01-19</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  pillar                     1.9.0     <span style="color: #555555;">2023-03-22</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  pkgconfig                  2.0.3     <span style="color: #555555;">2019-09-22</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  pkgdown                    2.0.7     <span style="color: #555555;">2022-12-14</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  plotly                     4.10.2    <span style="color: #555555;">2023-06-03</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  plyr                       1.8.8     <span style="color: #555555;">2022-11-11</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  png                        0.1-8     <span style="color: #555555;">2022-11-29</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  prettyunits                1.1.1     <span style="color: #555555;">2020-01-24</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  progress                   1.2.2     <span style="color: #555555;">2019-05-16</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  purrr                    * 1.0.1     <span style="color: #555555;">2023-01-10</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  qsvaR                    * 1.0.0     <span style="color: #555555;">2022-04-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  R6                         2.5.1     <span style="color: #555555;">2021-08-19</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  ragg                       1.2.5     <span style="color: #555555;">2023-01-12</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  rbibutils                  2.2.13    <span style="color: #555555;">2023-01-13</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  RColorBrewer               1.1-3     <span style="color: #555555;">2022-04-03</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  Rcpp                       1.0.11    <span style="color: #555555;">2023-07-06</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  RCurl                      1.98-1.12 <span style="color: #555555;">2023-03-27</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  Rdpack                     2.4       <span style="color: #555555;">2022-07-20</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  readr                    * 2.1.4     <span style="color: #555555;">2023-02-10</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  RefManageR               * 1.4.0     <span style="color: #555555;">2022-09-30</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  reshape2                   1.4.4     <span style="color: #555555;">2020-04-09</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  RhpcBLASctl                0.23-42   <span style="color: #555555;">2023-02-11</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  rjson                      0.2.21    <span style="color: #555555;">2022-01-09</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  rlang                      1.1.1     <span style="color: #555555;">2023-04-28</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  rmarkdown                  2.23      <span style="color: #555555;">2023-07-01</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  rprojroot                  2.0.3     <span style="color: #555555;">2022-04-02</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  RSQLite                    2.3.1     <span style="color: #555555;">2023-04-03</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  S4Vectors                * 0.34.0    <span style="color: #555555;">2022-04-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  sass                       0.4.7     <span style="color: #555555;">2023-07-15</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  scales                     1.2.1     <span style="color: #555555;">2022-08-20</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  sessioninfo              * 1.2.2     <span style="color: #555555;">2021-12-06</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  shape                      1.4.6     <span style="color: #555555;">2021-05-19</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  stringi                    1.7.12    <span style="color: #555555;">2023-01-11</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  stringr                  * 1.5.0     <span style="color: #555555;">2022-12-02</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  SummarizedExperiment     * 1.26.1    <span style="color: #555555;">2022-04-29</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  survival                   3.5-5     <span style="color: #555555;">2023-03-12</span> <span style="color: #555555;">[3]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  sva                        3.44.0    <span style="color: #555555;">2022-04-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  systemfonts                1.0.4     <span style="color: #555555;">2022-02-11</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  textshaping                0.3.6     <span style="color: #555555;">2021-10-13</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  tibble                   * 3.2.1     <span style="color: #555555;">2023-03-20</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  tidyr                    * 1.3.0     <span style="color: #555555;">2023-01-24</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  tidyselect                 1.2.0     <span style="color: #555555;">2022-10-10</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  tidySummarizedExperiment * 1.6.1     <span style="color: #555555;">2022-05-22</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  tidyverse                * 2.0.0     <span style="color: #555555;">2023-02-22</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  timechange                 0.2.0     <span style="color: #555555;">2023-01-11</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  tzdb                       0.4.0     <span style="color: #555555;">2023-05-12</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  utf8                       1.2.3     <span style="color: #555555;">2023-01-31</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  variancePartition        * 1.26.0    <span style="color: #555555;">2022-04-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  vctrs                      0.6.3     <span style="color: #555555;">2023-06-14</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  viridisLite                0.4.2     <span style="color: #555555;">2023-05-02</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  withr                      2.5.0     <span style="color: #555555;">2022-03-03</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  xfun                       0.39      <span style="color: #555555;">2023-04-20</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  XML                        3.99-0.14 <span style="color: #555555;">2023-03-19</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  xml2                       1.3.5     <span style="color: #555555;">2023-07-06</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  xtable                     1.8-4     <span style="color: #555555;">2019-04-21</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  XVector                    0.36.0    <span style="color: #555555;">2022-04-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  yaml                       2.3.7     <span style="color: #555555;">2023-01-23</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.2.0)</span></span></span>
<span><span class="co">#&gt;  zlibbioc                   1.42.0    <span style="color: #555555;">2022-04-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #555555;"> [1] /__w/_temp/Library</span></span></span>
<span><span class="co">#&gt; <span style="color: #555555;"> [2] /usr/local/lib/R/site-library</span></span></span>
<span><span class="co">#&gt; <span style="color: #555555;"> [3] /usr/local/lib/R/library</span></span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB; font-weight: bold;">──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────</span></span></span></code></pre>
</div>
<div class="section level2">
<h2 id="bibliography">Bibliography<a class="anchor" aria-label="anchor" href="#bibliography"></a>
</h2>
<p>This vignette was generated using <em><a href="https://bioconductor.org/packages/3.15/BiocStyle" class="external-link">BiocStyle</a></em>
(Oleś, 2022) with <em><a href="https://CRAN.R-project.org/package=knitr" class="external-link">knitr</a></em> (Xie,
2023) and <em><a href="https://CRAN.R-project.org/package=rmarkdown" class="external-link">rmarkdown</a></em>
(Allaire, Xie, Dervieux et al., 2023) running behind the scenes.</p>
<p>Citations made with <em><a href="https://CRAN.R-project.org/package=RefManageR" class="external-link">RefManageR</a></em>
(McLean, 2017).</p>
<p>
<a id="bib-allaire2023rmarkdown"></a><a href="#cite-allaire2023rmarkdown">[1]</a><cite>
J. Allaire, Y. Xie, C. Dervieux, et al. <em>rmarkdown: Dynamic Documents
for R</em>. R package version 2.23. 2023. URL:
<a href="https://github.com/rstudio/rmarkdown" class="external-link">https://github.com/rstudio/rmarkdown</a>.</cite>
</p>
<p>
<a id="bib-Law2020"></a><a href="#cite-Law2020">[2]</a><cite> W. L.
Charity, K. S. Gordon, E. R. Matthew, et al. “A guide to creating design
matrices for gene expression experiments”. In: <em>F1000Research</em> 9
(2020). ISSN: 1759796X. DOI:
<a href="https://doi.org/10.12688/f1000research.27893.1" class="external-link">10.12688/f1000research.27893.1</a>.</cite>
</p>
<p>
<a id="bib-ColladoTorres2019"></a><a href="#cite-ColladoTorres2019">[3]</a><cite>
L. Collado-Torres, E. E. Burke, A. Peterson, et al. “Regional
Heterogeneity in Gene Expression, Regulation, and Coherence in the
Frontal Cortex and Hippocampus across Development and Schizophrenia”.
In: <em>Neuron</em> 103 (2 Jul. 2019), pp. 203-216.e8. ISSN: 10974199.
DOI:
<a href="https://doi.org/10.1016/j.neuron.2019.05.013" class="external-link">10.1016/j.neuron.2019.05.013</a>.</cite>
</p>
<p>
<a id="bib-gu2022colorramp2"></a><a href="#cite-gu2022colorramp2">[4]</a><cite>
Z. Gu. <em>colorRamp2: Generate Color Mapping Functions</em>. R package
version 0.1.0. 2022. URL:
<a href="https://github.com/jokergoo/colorRamp2" class="external-link">https://github.com/jokergoo/colorRamp2</a>.</cite>
</p>
<p>
<a id="bib-gu2016complex"></a><a href="#cite-gu2016complex">[5]</a><cite>
Z. Gu, R. Eils, and M. Schlesner. “Complex heatmaps reveal patterns and
correlations in multidimensional genomic data”. In:
<em>Bioinformatics</em> (2016).</cite>
</p>
<p>
<a id="bib-hediatnani2023differential"></a><a href="#cite-hediatnani2023differential">[6]</a><cite>
HediaTnani. <em>Differential Expression Analysis using Limma and
qsvaR</em>. <a href="https://github.com/HediaTnani/qsvaRPkgDemo/qsvaRPkgDemo" class="external-link uri">https://github.com/HediaTnani/qsvaRPkgDemo/qsvaRPkgDemo</a>
- R package version 0.99.0. 2023. DOI:
<a href="https://doi.org/10.18129/B9.bioc.qsvaRPkgDemo" class="external-link">10.18129/B9.bioc.qsvaRPkgDemo</a>.
URL:
<a href="http://www.bioconductor.org/packages/qsvaRPkgDemo" class="external-link">http://www.bioconductor.org/packages/qsvaRPkgDemo</a>.</cite>
</p>
<p>
<a id="bib-hoffman2016variancepartition"></a><a href="#cite-hoffman2016variancepartition">[7]</a><cite>
G. E. Hoffman and E. E. Schadt. “variancePartition: Interpreting drivers
of variation in complex gene expression studies”. In: <em>BMC
Bioinformatics</em> 17 (483 2016). DOI:
<a href="https://doi.org/10.1186/s12859-016-1323-z" class="external-link">10.1186/s12859-016-1323-z</a>.</cite>
</p>
<p>
<a id="bib-Jaffe2017"></a><a href="#cite-Jaffe2017">[8]</a><cite> A. E.
Jaffe, R. Tao, A. L. Norris, et al. “QSVA framework for RNA quality
correction in differential expression analysis”. In: <em>Proceedings of
the National Academy of Sciences of the United States of America</em>
114 (27 Jul. 2017), pp. 7130-7135. ISSN: 10916490. DOI:
<a href="https://doi.org/10.1073/pnas.1617384114" class="external-link">10.1073/pnas.1617384114</a>.</cite>
</p>
<p>
<a id="bib-mangiola2022tidysummarizedexperiment"></a><a href="#cite-mangiola2022tidysummarizedexperiment">[9]</a><cite>
S. Mangiola. <em>tidySummarizedExperiment: Brings SummarizedExperiment
to the Tidyverse</em>. R package version 1.6.1. 2022. URL:
<a href="https://github.com/stemangiola/tidySummarizedExperiment" class="external-link">https://github.com/stemangiola/tidySummarizedExperiment</a>.</cite>
</p>
<p>
<a id="bib-mclean2017refmanager"></a><a href="#cite-mclean2017refmanager">[10]</a><cite>
M. W. McLean. “RefManageR: Import and Manage BibTeX and BibLaTeX
References in R”. In: <em>The Journal of Open Source Software</em>
(2017). DOI:
<a href="https://doi.org/10.21105/joss.00338" class="external-link">10.21105/joss.00338</a>.</cite>
</p>
<p>
<a id="bib-morgan2022summarizedexperiment"></a><a href="#cite-morgan2022summarizedexperiment">[11]</a><cite>
M. Morgan, V. Obenchain, J. Hester, et al. <em>SummarizedExperiment:
SummarizedExperiment container</em>. R package version 1.26.1. 2022.
URL:
<a href="https://bioconductor.org/packages/SummarizedExperiment" class="external-link">https://bioconductor.org/packages/SummarizedExperiment</a>.</cite>
</p>
<p>
<a id="bib-ole2022biocstyle"></a><a href="#cite-ole2022biocstyle">[12]</a><cite>
A. Oleś. <em>BiocStyle: Standard styles for vignettes and other
Bioconductor documents</em>. R package version 2.24.0. 2022. URL:
<a href="https://github.com/Bioconductor/BiocStyle" class="external-link">https://github.com/Bioconductor/BiocStyle</a>.</cite>
</p>
<p>
<a id="bib-2022language"></a><a href="#cite-2022language">[13]</a><cite>
R Core Team. <em>R: A Language and Environment for Statistical
Computing</em>. R Foundation for Statistical Computing. Vienna, Austria,
2022. URL:
<a href="https://www.R-project.org/" class="external-link">https://www.R-project.org/</a>.</cite>
</p>
<p>
<a id="bib-ritchie2015limma"></a><a href="#cite-ritchie2015limma">[14]</a><cite>
M. E. Ritchie, B. Phipson, D. Wu, et al. “limma powers differential
expression analyses for RNA-sequencing and microarray studies”. In:
<em>Nucleic Acids Research</em> 43.7 (2015), p. e47. DOI:
<a href="https://doi.org/10.1093/nar/gkv007" class="external-link">10.1093/nar/gkv007</a>.</cite>
</p>
<p>
<a id="bib-stolz2023qsvar"></a><a href="#cite-stolz2023qsvar">[15]</a><cite>
J. M. Stolz and L. Collado-Torres. <em>qsvaR</em>. <a href="https://github.com/LieberInstitute/qsvaR/qsvaR" class="external-link uri">https://github.com/LieberInstitute/qsvaR/qsvaR</a> - R
package version 1.0.0. 2023. DOI:
<a href="https://doi.org/10.18129/B9.bioc.qsvaR" class="external-link">10.18129/B9.bioc.qsvaR</a>.
URL:
<a href="http://www.bioconductor.org/packages/qsvaR" class="external-link">http://www.bioconductor.org/packages/qsvaR</a>.</cite>
</p>
<p>
<a id="bib-wickham2011testthat"></a><a href="#cite-wickham2011testthat">[16]</a><cite>
H. Wickham. “testthat: Get Started with Testing”. In: <em>The R
Journal</em> 3 (2011), pp. 5–10. URL:
<a href="https://journal.r-project.org/archive/2011-1/RJournal_2011-1_Wickham.pdf" class="external-link">https://journal.r-project.org/archive/2011-1/RJournal_2011-1_Wickham.pdf</a>.</cite>
</p>
<p>
<a id="bib-wickham2019welcome"></a><a href="#cite-wickham2019welcome">[17]</a><cite>
H. Wickham, M. Averick, J. Bryan, et al. “Welcome to the tidyverse”. In:
<em>Journal of Open Source Software</em> 4.43 (2019), p. 1686. DOI:
<a href="https://doi.org/10.21105/joss.01686" class="external-link">10.21105/joss.01686</a>.</cite>
</p>
<p>
<a id="bib-wickham2021sessioninfo"></a><a href="#cite-wickham2021sessioninfo">[18]</a><cite>
H. Wickham, W. Chang, R. Flight, et al. <em>sessioninfo: R Session
Information</em>. <a href="https://github.com/r-lib/sessioninfo#readme" class="external-link uri">https://github.com/r-lib/sessioninfo#readme</a>, <a href="https://r-lib.github.io/sessioninfo/" class="external-link uri">https://r-lib.github.io/sessioninfo/</a>. 2021.</cite>
</p>
<p>
<a id="bib-wickham2023dplyr"></a><a href="#cite-wickham2023dplyr">[19]</a><cite>
H. Wickham, R. François, L. Henry, et al. <em>dplyr: A Grammar of Data
Manipulation</em>. <a href="https://dplyr.tidyverse.org" class="external-link uri">https://dplyr.tidyverse.org</a>, <a href="https://github.com/tidyverse/dplyr" class="external-link uri">https://github.com/tidyverse/dplyr</a>. 2023.</cite>
</p>
<p>
<a id="bib-wickham2023purrr"></a><a href="#cite-wickham2023purrr">[20]</a><cite>
H. Wickham and L. Henry. <em>purrr: Functional Programming Tools</em>.
<a href="https://purrr.tidyverse.org/" class="external-link uri">https://purrr.tidyverse.org/</a>, <a href="https://github.com/tidyverse/purrr" class="external-link uri">https://github.com/tidyverse/purrr</a>. 2023.</cite>
</p>
<p>
<a id="bib-xie2023knitr"></a><a href="#cite-xie2023knitr">[21]</a><cite>
Y. Xie. <em>knitr: A General-Purpose Package for Dynamic Report
Generation in R</em>. R package version 1.43. 2023. URL:
<a href="https://yihui.org/knitr/" class="external-link">https://yihui.org/knitr/</a>.</cite>
</p>
</div>
<div class="section level2">
<h2 id="other-resources">Other resources<a class="anchor" aria-label="anchor" href="#other-resources"></a>
</h2>
<ol style="list-style-type: decimal">
<li><p><a href="https://bioconductor.org/packages/release/workflows/vignettes/RNAseq123/inst/doc/designmatrices.html" class="external-link uri">https://bioconductor.org/packages/release/workflows/vignettes/RNAseq123/inst/doc/designmatrices.html</a></p></li>
<li><p><a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7873980/" class="external-link uri">https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7873980/</a></p></li>
<li><p><a href="https://bioconductor.org/help/course-materials/2019/BSS2019/04_Practical_CoreApproachesInBioconductor.html" class="external-link uri">https://bioconductor.org/help/course-materials/2019/BSS2019/04_Practical_CoreApproachesInBioconductor.html</a></p></li>
<li><p><a href="https://jdblischak.github.io/dc-bioc-limma/linear-models.html" class="external-link uri">https://jdblischak.github.io/dc-bioc-limma/linear-models.html</a></p></li>
<li><p><a href="https://carpentries-incubator.github.io/high-dimensional-stats-r/02-high-dimensional-regression/index.html" class="external-link uri">https://carpentries-incubator.github.io/high-dimensional-stats-r/02-high-dimensional-regression/index.html</a></p></li>
<li><p><a href="https://mlu-explain.github.io/linear-regression/" class="external-link uri">https://mlu-explain.github.io/linear-regression/</a></p></li>
</ol>
</div>
<div class="section level2">
<h2 id="acknowledgements">Acknowledgements<a class="anchor" aria-label="anchor" href="#acknowledgements"></a>
</h2>
<ul>
<li><p>Leonardo Collado-Torres : guidance and support with the
deployment of this workshop package and website</p></li>
<li><p>Nicholas J. Eagles: support with the deployment of this workshop
package and website</p></li>
</ul>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Hedia Tnani.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
